
<link rel="stylesheet" href="/assets/highlight/styles/default.css">
<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;margin:5;background-color:#CFB53B; cursor:pointer;padding-right:50px">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="padding-left:50px;padding-top:10px;background-color:#CFB53B;">UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a>

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>

<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:160px">

<h1 id="alignment-to-read-counts--visualization-in-igv">Alignment to Read Counts &amp; Visualization in IGV</h1>

<p>This document assumes <a href="/data_reduction/preproc_htstream.html">preproc htstream</a> has been completed.</p>

<p><strong>IF</strong> for some reason it didn’t finish, is corrupted or you missed the session, you can copy over a completed copy</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cp -r /share/biocore/workshops/2019_August_RNAseq/HTS_testing /share/workshop/$USER/rnaseq_example/.
cp -r /share/biocore/workshops/2019_August_RNAseq/01-HTS_Preproc /share/workshop/$USER/rnaseq_example/.
cp  /share/biocore/workshops/2019_August_RNAseq/summary_hts.txt /share/workshop/$USER/rnaseq_example/.
</code></pre></div></div>

<h2 id="alignment-vs-assembly">Alignment vs Assembly</h2>

<p><strong>Given sequence data</strong>,</p>

<p><em>Assembly seeks to put together the puzzle without knowing what the picture is.</em></p>

<p><em>Mapping tries to put together the puzzle pieces directly onto an image of the picture.</em></p>

<p>In mapping the question is more, given a small chunk of sequence, where in the genome did this piece most likely come from.</p>

<p>The goal then is to find the match(es) with either the “best” edit distance (smallest), or all matches with edit distance less than max edit dist. Main issues are:</p>

<ul>
  <li>Large search space</li>
  <li>Regions of similarity (aka repeats)</li>
  <li>Gaps (INDELS)</li>
  <li>Complexity (RNA, transcripts)</li>
</ul>

<h4 id="considerations">Considerations</h4>
<ul>
  <li>Placing reads in regions that do not exist in the reference genome (reads extend off the end) [ mitochondrial, plasmids, structural variants, etc.].</li>
  <li>Sequencing errors and variations: alignment between read and true source in genome may have more differences than alignment with some other copy of repeat.</li>
  <li>What if the closest fully sequenced genome is too divergent? (3% is a common alignment capability)</li>
  <li>Placing reads in repetitive regions: Some algorithms only return 1 mapping; If multiple: map quality = 0</li>
  <li>Algorithms that use paired-end information =&gt; might prefer correct distance over correct alignment.</li>
</ul>

<p>In RNAseq data, you must also consider effect of splice junctions, reads may span an intron.</p>

<p><img src="alignment_figures/alignment_figure1.png" alt="alignment_figure1" width="800px" /></p>

<h3 id="aligners">Aligners</h3>
<p>Many <a href="https://en.wikipedia.org/wiki/List_of_sequence_alignment_software">alignment algorithms</a> to choose from.</p>
<ul>
  <li>Spliced Aligners
    <ul>
      <li>STAR</li>
      <li>HiSAT2 (formerly Tophat [Bowtie2])</li>
      <li>GMAP - GSNAP</li>
      <li>SOAPsplice</li>
      <li>MapSplice</li>
    </ul>
  </li>
  <li>Aligners that can ’clip’
    <ul>
      <li>bwa-mem</li>
      <li>Bowtie2 in local mode</li>
    </ul>
  </li>
</ul>

<h4 id="pseudo-aligners-salmon-and-kalisto">Pseudo-aligners (salmon and kalisto)</h4>
<ul>
  <li>Quasi-mapping</li>
  <li>Probabilistic</li>
  <li>Map to transcripts, not genome</li>
  <li>Does transcript quantifications (or gene)</li>
  <li>Blazing FAST and can run on most laptops</li>
  <li>Experience suggests differences between “traditional” mappers are in the low abundance genes.</li>
</ul>

<h3 id="mapping-against-the-genome-vs-transcriptome">Mapping against the genome vs transcriptome</h3>

<ul>
  <li>May seem intuitive to map RNAseq data to transcriptome, but it is not that simple.
    <ul>
      <li>Transcriptomes are rarely complete.</li>
      <li>Which transcript of a gene should you map to, canonical transcript?</li>
      <li>Shouldn’t map to all splice variants as these would show up as ‘multi-mappers’.</li>
    </ul>
  </li>
  <li>More so, a mapper will try to map every read, somewhere, provided the result meets its minimum requirements.
    <ul>
      <li>Need to provide a mapper with all possible places the read could have arisen from, which is best represented by the genome.</li>
      <li>Otherwise, you get mis-mapping because its close enough.</li>
    </ul>
  </li>
</ul>

<h3 id="genome-and-genome-annotation">Genome and Genome Annotation</h3>

<p>Genome sequence fasta files and annotation (gff, gtf) files go together! These should be identified at the beginning of analysis.</p>
<ul>
  <li>Genome fasta files should include all primary chromosomes, unplaced sequences and un-localized sequences, as well as any organelles. Should bet contain any contigs that represent patches, or alternative haplotypes.</li>
  <li>If you expect contamination, or the presence of additional sequence/genome, add the sequence(s) to the genome fasta file.</li>
  <li>Annotation file should be GTF (preferred), and should be the most comprehensive you can find.
    <ul>
      <li>Chromosome names in the GTF must match those in the fasta file (they don’t always do).</li>
      <li>Star recommends the Genecode annotations for mouse/human</li>
    </ul>
  </li>
</ul>

<h2 id="counting-reads-as-a-proxy-for-gene-expression">Counting reads as a proxy for gene expression</h2>

<p>The more you can count (and HTS sequencing systems can count a lot) the better the measure of copy number for even rare transcripts in a population.</p>
<ul>
  <li>Most RNA-seq techniques deal with count data. Reads are mapped to a reference genome, transcripts are detected, and the number of reads that map to a transcript (or gene) are counted.</li>
  <li>Read counts for a transcript are roughly proportional to the gene’s length and transcript abundance (in whole transcript methods).</li>
</ul>

<p>Technical artifacts should be considered during counting</p>
<ul>
  <li>Mapping quality</li>
  <li>Map-ability (uniqueness), the read is not ambiguous</li>
</ul>

<p>Options are (STAR, HTSEQ, featureCounts)</p>

<h3 id="the-htseq-way">The HTSEQ way</h3>
<p><strong>Problem:</strong></p>
<ul>
  <li>Given a sam/bam file with aligned sequence reads and a list of genomic feature (genes locations), we wish to count the number of reads (fragments) than overlap each feature.
    <ul>
      <li>Features are defined by intervals, they have a start and stop position on a chromosome.</li>
      <li>For this workshop and analysis, features are genes which are the union of all its exons. You could consider each exon as a feature, for alternative splicing.</li>
    </ul>
  </li>
  <li>Htseq-count has three overlapping modes
    <ul>
      <li>union</li>
      <li>intersection-strict</li>
      <li>intersection-nonempty</li>
    </ul>

    <p><img src="alignment_figures/alignment_figure2.png" alt="alignment_figure2" width="500px" /></p>
  </li>
  <li>from the htseq paper</li>
</ul>

<h4 id="star-implementation">Star Implementation</h4>
<p>Counts coincide with Htseq-counts under default parameters (union and tests all orientations). Need to specify GTF file at genome generation step or during mapping.</p>
<ul>
  <li>Output, 4 columns
    <ul>
      <li>GeneID</li>
      <li>Counts for unstranded</li>
      <li>Counts for first read strand</li>
      <li>Counts for second read strand</li>
    </ul>
  </li>
</ul>

<p>Choose the appropriate column given the library preparation characteristics and generate a matrix table, columns are samples, rows are genes.</p>

<h3 id="alignment-concepts">Alignment concepts</h3>

<ul>
  <li>Multimappers:
    <ul>
      <li>Reads that align equally well to more than one reference location.</li>
      <li>Generally, multimappers are discounted in variant detection, and are often discounted in counting
applications (like RNA-Seq … would “cancel” out anyway).</li>
      <li>Note: multimapper “rescue” in some algorithms (RSEM, Express?).</li>
    </ul>
  </li>
  <li>Duplicates:
    <ul>
      <li>Reads or read pairs arising from the same original library fragment, either during library preparation (PCR duplicates).</li>
      <li>Generally, duplicates can only be detected reliably with paired-end sequencing. If PE, they’re discounted in variant detection, and discounted in counting applications (like RNA-Seq).</li>
    </ul>
  </li>
  <li>Clipping vs Splicing<br />
<img src="alignment_figures/alignment_figure3.png" alt="alignment_figure3" width="400px" /></li>
  <li>Inner length, insert size, fragment length<br />
<img src="alignment_figures/alignment_figure4.jpg" alt="alignment_figure4" width="400px" /><br />
*From https://www.biostars.org/p/106291/</li>
</ul>

<h2 id="indexing-a-reference-sequence-and-annotation">Indexing a Reference sequence and annotation</h2>

<p><strong>1.</strong> First lets make sure we are where we are supposed to be and that the References directory is available.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example
</code></pre></div></div>

<hr />
<p><strong>2.</strong> To align our data we will need the genome (fasta) and annotation (gtf) for human. There are many places to find them, but we are going to get them from the <a href="https://www.gencodegenes.org/human/">GENCODE</a>.</p>

<p>We need to first get the url for the annotation gtf. For RNAseq we want to use the PRI (primary) genome chromosome and Basic gene annotation. At the time of this workshop the current version of GENCODE is <em>31</em>. You will want to update the scripts to use the current version.</p>

<p><img src="alignment_figures/index_figure1.png" alt="index_figure1" width="800px" /></p>

<p><img src="alignment_figures/index_figure2.png" alt="index_figure2" width="800px" /></p>

<p><strong>2.</strong> We are going to use an aligner called <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905/">‘STAR’</a> to align the data, but first we need to index the genome for STAR. Lets pull down a slurm script to index the human GENCODE version of the genome.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/star_index.slurm
less star_index.slurm
</code></pre></div></div>

<p>When you are done, type “q” to exit.</p>

<div class="script">#!/bin/bash

#SBATCH --job-name=star_index # Job name
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --time=120
#SBATCH --mem=40000 # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --partition=production
#SBATCH --reservation=workshop
#SBATCH --account=workshop
#SBATCH --output=slurmout/star-index_%A.out # File to which STDOUT will be written
#SBATCH --error=slurmout/star-index_%A.err # File to which STDERR will be written

start=`date +%s`
echo $HOSTNAME

outpath="References"
[[ -d ${outpath} ]] || mkdir ${outpath}

cd ${outpath}
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_31/GRCh38.primary_assembly.genome.fa.gz
gunzip GRCh38.primary_assembly.genome.fa.gz
FASTA="../GRCh38.primary_assembly.genome.fa"

wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_31/gencode.v31.primary_assembly.annotation.gtf.gz
gunzip gencode.v31.primary_assembly.annotation.gtf.gz
GTF="../gencode.v31.primary_assembly.annotation.gtf"

mkdir star.overlap100.gencode.v31
cd star.overlap100.gencode.v31

module load star/2.7.0e

call="STAR
     --runThreadN 8 \
     --runMode genomeGenerate \
     --genomeDir . \
     --sjdbOverhang 100 \
     --sjdbGTFfile ${GTF} \
     --genomeFastaFiles ${FASTA}"

echo $call
eval $call

end=`date +%s`
runtime=$((end-start))
echo $runtime
</div>

<ol>
  <li>The script uses wget to download the fasta and GTF files from GENCODE using the links you found earlier.</li>
  <li>Uncompresses them using gunzip.</li>
  <li>Creates the star index directory [star.overlap100.gencode.v31].</li>
  <li>Change directory into the new star index directory. We run the star indexing command from inside the directory, for some reason star fails if you try to run it outside this directory.</li>
  <li>Run star in mode genomeGenerate.</li>
</ol>

<p>Run star indexing when ready.</p>

<p>sbatch star_index.slurm</p>

<p>This step will take a couple hours. You can look at the <a href="https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf">STAR documentation</a> while you wait. All of the output files will be written to the star_index directory.</p>

<p><strong>IF</strong> For the sake of time, or for some reason it didn’t finish, is corrupted, or you missed the session, you can link over a completed copy.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ln -s /share/biocore/workshops/2019_August_RNAseq/References/star.overlap100.gencode.v31 /share/workshop/$USER/rnaseq_example/References/.
</code></pre></div></div>

<h2 id="alignments">Alignments</h2>

<p><strong>1.</strong> We are now ready to try an alignment:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example/HTS_testing
</code></pre></div></div>

<p>and let’s run STAR (via srun) on the pair of streamed test files we created earlier:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>srun --time=15:00:00 -n 8 --mem=32g --reservation=workshop --account=workshop --pty /bin/bash
</code></pre></div></div>

<p>Once you’ve been given an interactive session we can run STAR. You can ignore the two warnings/errors and you know your on a cluster node because your server will change. Here you see I’m on tadpole, then after the srun command is successful, I am now on drove-13.</p>

<div class="output">msettles@tadpole:/share/workshop/msettles/rnaseq_example/&gt; HTS_testing$ srun --time=15:00:00 -n 8 --mem=32g --reservation=workshop --account=workshop --pty /bin/bash
srun: job 29372920 queued and waiting for resources
srun: job 29372920 has been allocated resources
groups: cannot find name for group ID 2020
bash: /home/msettles/.bashrc: Permission denied
msettles@drove-13:/share/workshop/msettles/rnaseq_example/&gt; HTS_testing$
</div>

<p>Then run the star commands</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load star/2.7.0e
STAR \
--runThreadN 8 \
--genomeDir ../References/star.overlap100.gencode.v31 \
--outSAMtype BAM SortedByCoordinate \
--quantMode GeneCounts \
--outFileNamePrefix SampleAC1.streamed_ \
--readFilesCommand zcat \
--readFilesIn SampleAC1.streamed_R1.fastq.gz SampleAC1.streamed_R2.fastq.gz
</code></pre></div></div>

<p>In the command, we are telling star to count reads on a gene level (‘–quantMode GeneCounts’), the prefix for all the output files will be SampleAC1.streamed_, the command to unzip the files (zcat), and finally, the input file pair.</p>

<p>Once finished please ‘exit’ the srun session. You’ll know you were successful when your back on tadpole</p>

<div class="output">msettles@drove-13:/share/workshop/msettles/rnaseq_example/HTS_testing$ exit
exit
msettles@tadpole:/share/workshop/msettles/rnaseq_example/HTS_testing$
</div>

<h2 id="now-lets-take-a-look-at-an-alignment-in-igv">Now let’s take a look at an alignment in IGV.</h2>

<p><strong>1.</strong> We first need to index the bam file, will use ‘samtools’ for this step, which is a program to manipulate SAM/BAM files. Take a look at the options for samtools and ‘samtools index’.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load samtools/1.9
samtools
samtools index
</code></pre></div></div>

<p>We need to index the BAM file:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example/HTS_testing
samtools index SampleAC1.streamed_Aligned.sortedByCoord.out.bam
</code></pre></div></div>

<p><strong>IF</strong> for some reason it didn’t finish, is corrupted or you missed the session, you can copy over a completed copy</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cp -r /share/biocore/workshops/2019_August_RNAseq/HTS_testing/SampleAC1.streamed_Aligned.sortedByCoord.out.bam* /share/workshop/$USER/rnaseq_example/HTS_testing
</code></pre></div></div>

<hr />
<p><strong>2.</strong> Transfer SampleAC1.streamed_Aligned.sortedByCoord.out.bam and SampleAC1.streamed_Aligned.sortedByCoord.out.bam (the index file) to your computer using scp or winSCP, or copy/paste from cat [sometimes doesn’t work].</p>

<p>In a new shell session on my laptop. <strong>NOT logged into tadpole</strong>.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir ~/rnaseq_workshop
cd ~/rnaseq_workshop
scp msettles@tadpole.genomecenter.ucdavis.edu:/share/workshop/msettles/rnaseq_example/HTS_testing/SampleAC1.streamed_Aligned.sortedByCoord.out.bam* .
</code></pre></div></div>

<p>Its ok of the mkdir command fails (“File exists”) because we aleady created the directory earlier.</p>

<hr />
<p><strong>3.</strong> Now we are ready to use IGV. Go to the <a href="http://software.broadinstitute.org/software/igv/">IGV page at the Broad Institute</a>.</p>

<p><img src="alignment_figures/index_igv1.png" alt="index_igv1" width="800px" /></p>

<p>And then navigate to the download page, <a href="http://software.broadinstitute.org/software/igv/download">IGV download</a></p>

<p><img src="alignment_figures/index_igv2.png" alt="index_igv2" width="800px" /></p>

<p>Here you can download IGV for your respective platform (Window, Mac OSX, Linux), but we are going to use the web application they supply, <a href="https://igv.org/app/">IGV web app</a>.</p>

<p><img src="alignment_figures/index_igv3.png" alt="index_igv3" width="800px" /></p>

<hr />
<p><strong>4.</strong> The first thing we want to do is load the Human genome. Click on “Genomes” in the menu and choose “Human (GRCh38/hg38)”.</p>

<p><img src="alignment_figures/index_igv4.png" alt="index_igv4" width="800px" /></p>

<hr />
<p><strong>5.</strong> Now let’s load the alignment bam and index files. Click on “Tracks” and choose “Local File …”.</p>

<p><img src="alignment_figures/index_igv5.png" alt="index_igv5" width="800px" /></p>

<p>Navigate to where you transferred the bam and index file and select them both.</p>

<p><img src="alignment_figures/index_igv6.png" alt="index_igv6" width="800px" /></p>

<p>Now your alignment is loaded. Any loaded bam file aligned to a genome is called a “track”.</p>

<p><img src="alignment_figures/index_igv7.png" alt="index_igv7" width="800px" /></p>

<hr />
<p><strong>6.</strong> Lets take a look at the alignment associated with the gene <strong>HBB</strong>, and if for some reason it doesn’t find HBB (web IGV can be fickle) go to position <strong>chr11:5,224,466-5,228,071</strong></p>

<p><img src="alignment_figures/index_igv8.png" alt="index_igv8" width="800px" /></p>

<p><img src="alignment_figures/index_igv9.png" alt="index_igv9" width="800px" /></p>

<p>You can zoom in by clicking on the plus sign (top right) or zoom out by clicking the negative sign. You also may have to move around by clicking and dragging in the BAM track window.</p>

<p>You can also zoom in by clicking and dragging across the number line at the top. That section will highlight, and when you release the button, it will zoom into that section.</p>

<p><img src="alignment_figures/index_igv10.png" alt="index_igv10" width="800px" /></p>

<p><img src="alignment_figures/index_igv11.png" alt="index_igv11" width="800px" /></p>

<p>Reset the window by searching for HBB again. And zoom in 1 step.</p>

<p><img src="alignment_figures/index_igv12.png" alt="index_igv12" width="800px" /></p>

<hr />
<p><strong>7.</strong> See that the reads should be aligning within the exons in the gene. This makes sense, since RNA-Seq reads are from exons. Play with the settings on the right hand side a bit.</p>

<p><img src="alignment_figures/index_igv13.png" alt="index_igv13" width="800px" /></p>

<p><img src="alignment_figures/index_igv14.png" alt="index_igv14" width="800px" /></p>

<p><img src="alignment_figures/index_igv15.png" alt="index_igv15" width="800px" /></p>

<p><img src="alignment_figures/index_igv16.png" alt="index_igv16" width="800px" /></p>

<p><img src="alignment_figures/index_igv17.png" alt="index_igv17" width="800px" /></p>

<hr />

<h2 id="running-star-on-the-experiment">Running STAR on the experiment</h2>

<p><strong>1.</strong> We can now run STAR across all samples on the real data using a SLURM script, <a href="/scripts/star.slurm">star.slurm</a>, that we should take a look at now.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example  # We'll run this from the main directory
wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/star.slurm
less star.slurm
</code></pre></div></div>

<p>When you are done, type “q” to exit.</p>

<pre class="prettyprint"><code class="language-sh" style="background-color:333333">
#!/bin/bash

#SBATCH --job-name=star # Job name
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --time=60
#SBATCH --mem=32000 # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --partition=production
#SBATCH --reservation=workshop
#SBATCH --account=workshop
#SBATCH --array=1-16
#SBATCH --output=slurmout/star_%A_%a.out # File to which STDOUT will be written
#SBATCH --error=slurmout/star_%A_%a.err # File to which STDERR will be written

start=`date +%s`
echo $HOSTNAME
echo "My SLURM_ARRAY_TASK_ID: " $SLURM_ARRAY_TASK_ID

sample=`sed "${SLURM_ARRAY_TASK_ID}q;d" samples.txt`
REF="References/star.overlap100.gencode.v31"

outpath='02-STAR_alignment'
[[ -d ${outpath} ]] || mkdir ${outpath}
[[ -d ${outpath}/${sample} ]] || mkdir ${outpath}/${sample}

echo "SAMPLE: ${sample}"

module load star/2.7.0e

call="STAR
     --runThreadN 8 \
     --genomeDir $REF \
     --outSAMtype BAM SortedByCoordinate \
     --readFilesCommand zcat \
     --readFilesIn 01-HTS_Preproc/${sample}/${sample}_R1.fastq.gz 01-HTS_Preproc/${sample}/${sample}_R2.fastq.gz \
     --quantMode GeneCounts \
     --outFileNamePrefix ${outpath}/${sample}/${sample}_ \
     ${outpath}/${sample}/${sample}-STAR.stdout 2&gt; ${outpath}/${sample}/${sample}-STAR.stderr"

echo $call
eval $call

end=`date +%s`
runtime=$((end-start))
echo $runtime
</code></pre>

<p>After looking at the script, lets run it.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sbatch star.slurm  # moment of truth!
</code></pre></div></div>

<p>We can watch the progress of our task array using the ‘squeue’ command. Takes about 30 minutes to process each sample.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>squeue -u msettles  # use your username
</code></pre></div></div>

<h2 id="quality-assurance---mapping-statistics-as-qaqc">Quality Assurance - Mapping statistics as QA/QC.</h2>

<p><strong>1.</strong> Once your jobs have finished successfully (check the error and out logs like we did in the previous exercise), use a script of ours, <a href="/scripts/star_stats.sh">star_stats.sh</a> to collect the alignment stats. Don’t worry about the script’s contents at the moment; you’ll use very similar commands to create a counts table in the next section. For now:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example  # We'll run this from the main directory
wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/star_stats.sh
bash star_stats.sh
</code></pre></div></div>

<p><strong>2.</strong> Transfer summary_star_alignments.txt to your computer using scp or winSCP, or copy/paste from cat [sometimes doesn’t work],</p>

<p>In a new shell session on your laptop. <strong>NOT logged into tadpole</strong>.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir ~/rnaseq_workshop
cd ~/rnaseq_workshop
scp [your_username]@tadpole.genomecenter.ucdavis.edu:/share/workshop/[your_username]/rnaseq_example/summary_star_alignments.txt .
</code></pre></div></div>

<p>Its ok of the mkdir command fails (“File exists”) because we aleady created the directory earlier.</p>

<p>Open in excel (or excel like application), and lets review.</p>

<p>The table that this script creates (“summary_star_alignments.txt”) can be pulled to your laptop via ‘scp’, or WinSCP, etc., and imported into a spreadsheet. Are all samples behaving similarly? Discuss …</p>

<h2 id="scripts">Scripts</h2>

<p>slurm script for indexing the genome</p>

<p><a href="/scripts/star_index.slurm">star_index.slurm</a></p>

<p>shell script for indexing the genome</p>

<p><a href="/scripts/star_index.sh">star_index.sh</a></p>

<p>slurm script for mapping using slurm task array and star</p>

<p><a href="/scripts/star.slurm">star.slurm</a></p>

<p>shell script for mapping using bash loop and star.</p>

<p><a href="/scripts/star.sh">star.sh</a></p>

<p>shell script to produce summary mapping table</p>

<p><a href="/scripts/star_stats.sh">star_stats.sh</a></p>


</div>



<script src="/assets/js/scale.fix.js"></script>



<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'true', 'auto');
    ga('send', 'pageview');
</script>


<script>


var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};
//var scrollpos=0;
//document.getElementById("mySidenav").onscroll = function() {scrollpos = document.getElementById("mySidenav").scrollTop;}

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
