
<head>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-146433201-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-146433201-1');

</script>
</head>


<link rel="stylesheet" href="/assets/highlight/styles/default.css">

<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>

<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="https://bioinformatics.ucdavis.edu/training/" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;padding-left:10px;padding-top:8px;background-color:#CFB53B; cursor:pointer;">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="border-radius: 0px;padding-top:10px;background-color:#CFB53B;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <!-- <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a> -->

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>


<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:170px">

<h1 id="sequence-preprocessing">Sequence preprocessing</h1>

<p>This document assumes <a href="/data_reduction/project_setup.html">project_setup</a> has been completed.</p>

<h2 id="why-preprocess-reads">Why preprocess reads</h2>

<p>We have found that aggressively “cleaning” and processing reads can make a large difference to the speed and quality of mapping and assembly results. Cleaning your reads means, removing reads/bases that are:</p>
<ul>
  <li>Unwanted sequence (Ex. polyA tails in RNAseq data)</li>
  <li>artificially added onto sequence of primary interest (vectors, adapters, primers)</li>
  <li>join short overlapping paired-end reads</li>
  <li>low quality bases</li>
  <li>originate from PCR duplication</li>
  <li>not of primary interest (contamination)</li>
</ul>

<p>Preprocessing also produces a number of statistics that are technical in nature that should be used to evaluate “experimental consistency”.</p>

<p><strong>Many read preprocessing strategies over time</strong></p>

<ul>
  <li>Identify and remove contaminant and vector reads.
    <ul>
      <li>Reads which appear to fully come from extraneous sequence should be removed.</li>
    </ul>
  </li>
  <li>Quality trim/cut.
    <ul>
      <li>“End” trim a read until the average quality &gt; Q (Lucy).</li>
      <li>Remove any read with average quality &lt; Q.</li>
    </ul>
  </li>
  <li>Eliminate singletons/duplicates.
    <ul>
      <li>If you have excess depth of coverage, and particularly if you have » x-fold coverage where x is the read length, then eliminating singletons is a nice way of dramatically reducing the number of candidate error-prone reads.</li>
      <li>Reads which appear the same (particularly paired-end) are often more likely PCR duplicates and therefor redundant reads.</li>
    </ul>
  </li>
  <li>Eliminate all reads (pairs) containing an “N” character.
    <ul>
      <li>If you can afford the loss of coverage, you might throw away all reads containing Ns.</li>
    </ul>
  </li>
  <li>Identify and trim off adapter and barcodes if present.
    <ul>
      <li>Believe it or not, the software provided by Illumina, either does not look for (default), or does a mediocre job of, identifying adapters and removing them.</li>
    </ul>
  </li>
</ul>

<p>** Many technical things happen between original sample and data, preprocessing is working backwards through that process to get as close as we can to original sample **</p>

<p><img src="preproc_figures/preproc_flowchart.png" alt="preproc_flowchart" width="800px" /></p>

<h2 id="rnaseq-preprocessing-workflow">RNAseq preprocessing workflow</h2>

<ol>
  <li>Remove contaminants (at least PhiX).</li>
  <li>Remove PCR duplicates.</li>
  <li>Identify rRNA proportion.</li>
  <li>Join and potentially extend, overlapping paired end reads</li>
  <li>If reads completely overlap they will contain adapter, remove adapters</li>
  <li>Identify and remove any adapter dimers present</li>
  <li>Trim sequences (5’ and 3’) by quality score (I like Q20)</li>
  <li>Cleanup
    <ul>
      <li>Remove any reads that are less then the minimum length parameter</li>
      <li>Run a polyA/T trimmer (optional)</li>
      <li>Produce preprocessing statistics</li>
    </ul>
  </li>
</ol>

<h2 id="htstream---preprocessing-application">HTStream - preprocessing application</h2>

<p>Can be downloaded from <a href="https://github.com/ibest/HTStream">here</a>. Fast C++ implementation, designed to have discreet applications that can be pipelined together using unix piping. We hope in the long run to include any and all needed preprocessing routines. Includes:</p>

<ul>
  <li>hts_AdapterTrimmer - identify and remove adapter sequences</li>
  <li>hts_NTrimmer - extract the longest subsequence with no Ns</li>
  <li>hts_PolyATTrim - identify and remove polyA/T sequence (least robust algorithm)</li>
  <li>hts_SeqScreener - identify and remove/keep/count contaminants (default phiX)</li>
  <li>hts_SuperDeduper - identify and remove PCR duplicates</li>
  <li>hts_CutTrim - discreet 5’ and/or 3’ basepair trimming</li>
  <li>hts_Overlapper - Overlap paired end reads (cutting off adapters when present)</li>
  <li>hts_QWindowTrim - 5’ and/or 3’ prime quality score trimming using windows</li>
  <li>hts_Stats - compute read stats</li>
</ul>

<p><strong>1.</strong> Let’s run the first step of our HTStream preprocessing pipeline, which is always to gather basic stats on the read files. For now, we’re only going to run one sample through the pipeline.</p>

<p><strong>1a.</strong> So let’s first take a small subsample of reads, just so our trial run through the pipeline goes really quickly.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example
mkdir HTS_testing
cd HTS_testing
zcat ../00-RawData/SampleAC1/SampleAC1_L3_R1.fastq.gz | head -400000 | gzip &gt; SampleAC1.subset_R1.fastq.gz
zcat ../00-RawData/SampleAC1/SampleAC1_L3_R2.fastq.gz | head -400000 | gzip &gt; SampleAC1.subset_R2.fastq.gz
ls
</code></pre></div></div>

<p>So we zcat (uncompress and send to screen), pipe to head (param -400000) then pipe to gzip to recompress and name our files subset.</p>

<p><em>How many reads are we going to analyze in our subset?</em></p>

<p><strong>1b.</strong> Now we’ll run our first preprocessing step … hts_Stats, first loading the module and then looking at help.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example/HTS_testing
module load htstream/1.1.0
hts_Stats --help
</code></pre></div></div>

<div class="output">HTStream &lt; https://github.com/ibest/HTStream &gt; application: hts_Stats
Version: 1.1.0-release
The hts_Stats app produce basic statistics about the reads in a dataset.
  Including the basepair composition and number of bases Q30.

Standard Options:
  -v [ --version ]                      Version print
  -h [ --help ]                         Prints help documentation
  -N [ --notes ] arg                    Notes for the stats JSON
  -L [ --stats-file ] arg (=stats.log)  String for output stats file name
  -A [ --append-stats-file ]            Append to stats file

Input Options [default: tab6 format on stdin]:
  -1 [ --read1-input ] arg              Read 1 paired end fastq input &lt; space
                                        separated for multiple files &gt;
  -2 [ --read2-input ] arg              Read 2 paired end fastq input &lt; space
                                        separated for multiple files &gt;
  -U [ --singleend-input ] arg          Single end read fastq input &lt; space
                                        separated for multiple files &gt;
  -T [ --tab-input ] arg                Tab-delimited (tab6) input &lt; space
                                        separated for multiple files &gt;
  -I [ --interleaved-input ] arg        Interleaved fastq input &lt; space
                                        separated for multiple files &gt;
  -S [ --from-stdin ]                   DEPRECATED PARAMETER

Output Options [default: tab6 format to stdout]:
  -F [ --force ]                        Forces overwrite of files
  -u [ --uncompressed ]                 Output uncompressed (not gzipped) files
  -f [ --fastq-output ] arg             Output to Fastq files &lt; PE AND/OR SE
                                        files &gt;
  -i [ --interleaved-output ] arg       Output to interleaved fastq files
                                        &lt; INTERLEAVED PE AND/OR SE files &gt;
  -t [ --tab-output ] arg               Output to tab-delimited (tab6) file
  -z [ --unmapped-output ] arg          Output to unmapped sam file
  -p [ --prefix ]                       DEPRECATED PARAMETER
  -g [ --gzip-output ]                  DEPRECATED PARAMETER
  -O [ --to-stdout ]                    DEPRECATED PARAMETER


Please report any issues, request for enhancement, or comments to &lt; https://github.com/ibest/HTStream/issues &gt;
</div>

<p>So now lets run hts_Stats and look at the output.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hts_Stats -1 SampleAC1.subset_R1.fastq.gz \
          -2 SampleAC1.subset_R2.fastq.gz \
          -L SampleAC1.stats.log -f SampleAC1.stats
</code></pre></div></div>

<p><em>What parameters did we use, what do they do?</em></p>

<p>Lets take a look at the output of stats</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ls -lah
</code></pre></div></div>

<div class="output">msettles@tadpole:/share/workshop/msettles/rnaseq_example/HTS_testing$ls -lah
total 29M
drwxrwsr-x 2 msettles workshop    8 Mar  9 10:46 .
drwxrwsr-x 5 msettles workshop    6 Mar  9 10:42 ..
-rw-rw-r-- 1 msettles workshop    0 Mar  9 10:45 C61.stats.log
-rw-rw-r-- 1 msettles workshop  660 Mar  9 10:46 SampleAC1.stats.log
-rw-rw-r-- 1 msettles workshop 6.8M Mar  9 10:46 SampleAC1.stats_R1.fastq.gz
-rw-rw-r-- 1 msettles workshop 7.5M Mar  9 10:46 SampleAC1.stats_R2.fastq.gz
-rw-rw-r-- 1 msettles workshop 6.8M Mar  9 10:42 SampleAC1.subset_R1.fastq.gz
-rw-rw-r-- 1 msettles workshop 7.5M Mar  9 10:42 SampleAC1.subset_R2.fastq.gz
</div>

<p><em>Which files were generated from hts_Stats?</em></p>

<p><em>Lets look at the file SampleAC1.stats.log</em></p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cat SampleAC1.stats.log
</code></pre></div></div>

<p>The logs generated by htstream are in <a href="https://en.wikipedia.org/wiki/JSON">JSON</a> format, like a database format but meant to be readable.</p>

<div class="output">{ "hts_Stats_63322": {
    "Notes": "",
    "totalFragmentsInput": 100000,
    "totalFragmentsOutput": 100000,
    "R1_readlength_histogram": [ [151,100000] ],
    "R2_readlength_histogram": [ [151,100000] ],
    "Base_composition": {
        "A": 7410244,
        "C": 7646646,
        "G": 7854214,
        "T": 7286488,
        "N": 2408
    },
    "Single_end": {
        "SE_in": 0,
        "SE_out": 0,
        "SE_bpLen": 0,
        "SE_bQ30": 0
    },
    "Paired_end": {
        "PE_in": 100000,
        "PE_out": 100000,
        "R1_bpLen": 15100000,
        "R1_bQ30": 14481157,
        "R2_bpLen": 15100000,
        "R2_bQ30": 13852279
    }
  }
}
</div>

<hr />
<p><strong>2.</strong> Next we are going to screen from ribosomal RNA (rRNA).</p>

<p>Ribosomal RNA can make up 90% or more of a typical <em>total RNA</em> sample. Most library prep methods attempt to reduce the rRNA representation in a sample, oligoDt binds to polyA tails to enrich a sample for mRNA, where Rabo-Depletion binds rRNA sequences to deplete the sample of rRNA. Neither technique is 100% efficient and so knowing the relative proportion of rRNA in each sample can be helpful.</p>

<p>Can screen for rRNA in our sample to determine rRNA efficiency.</p>

<p><strong>2a.</strong> Before we do so we need to find sequences of ribosomal RNA to screen against.</p>

<p>We will use these sequences to identify rRNA in our reads, which are from human. One way to do that is to go to <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> and search for them.</p>

<p><em>1.) First, go to NCBI and in the Search dropdown select “Taxonomy” and search for “human”.</em></p>

<p><img src="preproc_figures/ncbi01.png" alt="ncbi1" width="800px" /></p>

<p><em>2.) Click on “Homo sapiens”.</em></p>

<p><img src="preproc_figures/ncbi02.png" alt="ncbi2" width="800px" /></p>

<p><em>3.) Click on “Homo sapiens” again.</em></p>

<p><img src="preproc_figures/ncbi03.png" alt="ncbi3" width="800px" /></p>

<p><em>4.) Click on the “Subtree links” for Nucleotide</em></p>

<p><img src="preproc_figures/ncbi04.png" alt="ncbi4" width="800px" /></p>

<p><em>5.) Under Molecule Types, click on “rRNA” (left hand side).</em></p>

<p><img src="preproc_figures/ncbi05.png" alt="ncbi5" width="800px" /></p>

<p><em>6.) Click on “Send”, choose “File”, choose Format “FASTA”, and click on “Create File”.</em></p>

<p><img src="preproc_figures/ncbi06.png" alt="ncbi6" width="800px" /></p>

<p><img src="preproc_figures/ncbi07.png" alt="ncbi7" width="800px" /></p>

<p>Save this file to your computer, and rename it to ‘human_rrna.fasta’.</p>

<p>Now, make a directory in your “rnaseq_example” directory called “References”:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir /share/workshop/$USER/rnaseq_example/References
</code></pre></div></div>

<p>Upload your human_rrna.fasta file <strong>to the ‘References’ directory</strong> on the cluster using either <strong>scp</strong> or FileZilla. Or if you feel like ‘cheating’, just copy/paste the contents of human_rrna.fa using nano into a file named /share/workshop/$USER/rnaseq_example/References/human_rrna.fasta</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>nano /share/workshop/$USER/rnaseq_example/References/human_rrna.fasta
</code></pre></div></div>

<p>paste contents of human_rrna.fa and save</p>

<hr />

<p><strong>2b.</strong> Lets look for ribosomal rna, but not remove, so just count the occurrences.</p>

<p>First, view the help documentation for hts_SeqScreener</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example/HTS_testing
hts_SeqScreener -h
</code></pre></div></div>

<div class="output">HTStream &lt; https://github.com/ibest/HTStream &gt; application: hts_SeqScreener
Version: 1.1.0-release
hts_SeqScreener identifies and removes any reads which appear to have originated
  from a contaminant DNA source. Because bacteriophage Phi-X is common spiked
  into Illumina runs for QC purposes, sequences originating from Phi-X are removed
  by default. If other contaminants are suspected their sequence can be supplied
  as a fasta file &lt; seq&gt;, however the algorithm has been tuned for short contaminant
  sequences, and may not work well with sequences significantly longer than Phi-X (5Kb).


Standard Options:
  -v [ --version ]                      Version print
  -h [ --help ]                         Prints help documentation
  -N [ --notes ] arg                    Notes for the stats JSON
  -L [ --stats-file ] arg (=stats.log)  String for output stats file name
  -A [ --append-stats-file ]            Append to stats file

Input Options [default: tab6 format on stdin]:
  -1 [ --read1-input ] arg              Read 1 paired end fastq input &lt; space
                                        separated for multiple files &gt;
  -2 [ --read2-input ] arg              Read 2 paired end fastq input &lt; space
                                        separated for multiple files &gt;
  -U [ --singleend-input ] arg          Single end read fastq input &lt; space
                                        separated for multiple files &gt;
  -T [ --tab-input ] arg                Tab-delimited (tab6) input &lt; space
                                        separated for multiple files &gt;
  -I [ --interleaved-input ] arg        Interleaved fastq input &lt; space
                                        separated for multiple files &gt;
  -S [ --from-stdin ]                   DEPRECATED PARAMETER

Output Options [default: tab6 format to stdout]:
  -F [ --force ]                        Forces overwrite of files
  -u [ --uncompressed ]                 Output uncompressed (not gzipped) files
  -f [ --fastq-output ] arg             Output to Fastq files &lt; PE AND/OR SE
                                        files &gt;
  -i [ --interleaved-output ] arg       Output to interleaved fastq files
                                        &lt; INTERLEAVED PE AND/OR SE files &gt;
  -t [ --tab-output ] arg               Output to tab-delimited (tab6) file
  -z [ --unmapped-output ] arg          Output to unmapped sam file
  -p [ --prefix ]                       DEPRECATED PARAMETER
  -g [ --gzip-output ]                  DEPRECATED PARAMETER
  -O [ --to-stdout ]                    DEPRECATED PARAMETER

Application Specific Options:
  -s [ --seq ] arg                      Please supply a fasta file - default -
                                        Phix Sequence - default
                                        https://www.ncbi.nlm.nih.gov/nuccore/96
                                        26372
  -C [ --check-read-2 ]                 Check R2 as well as R1 (pe)
  -k [ --kmer ] arg (=12)               Kmer size of the lookup table (min 5,
                                        max 256)
  -x [ --percentage-hits ] arg (=0.25)  Proportion of kmer percentage-hits to
                                        sequence need to happen to discard (min
                                        0.0, max 1.0)
  -n [ --inverse ]                      Output reads that are ABOVE the kmer
                                        hit threshold
  -r [ --record ]                       Only record the reads that pass the
                                        kmer hit threshold, output all reads


Please report any issues, request for enhancement, or comments to &lt; https://github.com/ibest/HTStream/issues &gt;
</div>

<p><em>What parameters are needed to 1) provide a reference to hts_SeqScreener and 2) count, and not screen occurrences?</em></p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hts_SeqScreener -1 SampleAC1.subset_R1.fastq.gz \
                -2 SampleAC1.subset_R2.fastq.gz \
                -s ../References/human_rrna.fasta -r -L SampleAC1.rrna.log -f SampleAC1.rrna
</code></pre></div></div>

<p><em>Which files were generated from hts_SeqScreener?</em></p>

<p><em>Lets look at the file SampleAC1.rrna.log?</em></p>

<div class="output">{ "hts_SeqScreener_56678": {
    "Notes": "",
    "totalFragmentsInput": 100000,
    "totalFragmentsOutput": 100000,
    "inverse": 0,
    "record": 1,
    "Screen_info": {
        "screenBP": 107854,
        "lookupKmers": 16765
    },
    "Single_end": {
        "SE_in": 0,
        "SE_out": 0,
        "SE_hits": 0
    },
    "Paired_end": {
        "PE_in": 100000,
        "PE_out": 100000,
        "PE_hits": 3221
    }
  }
}
</div>

<p><em>What do you notice about the SampleAC1.rrna.log?</em></p>

<p><em>How many reads were identified as rrna?</em></p>

<p><strong>3.</strong> The power of htstream is the ability to stream reads through multiple programs using pipe’s.</p>

<p>By streaming reads through programs processing will be much quicker, each read is read in only once (written only once), and use significantly less storage as there are no intermediate files. It can do this by streaming a tab-delimited format called tab6.</p>

<p>Single end reads are 3 columns:</p>

<p><code class="highlighter-rouge">read1id  read1seq  read1qual</code></p>

<p>Paired end reads are 6 columns:</p>

<p><code class="highlighter-rouge">read1id  read1seq  read1qual  read2id  read2seq  read2qual</code></p>

<p>So lets first run hts_Stats and then hts_SeqScreener in a streamed fashion.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example/HTS_testing

hts_Stats -1 SampleAC1.subset_R1.fastq.gz \
          -2 SampleAC1.subset_R2.fastq.gz \
          -L SampleAC1.streamed.log |
hts_SeqScreener -A -L SampleAC1.streamed.log \
          -r -s ../References/human_rrna.fasta -f SampleAC1.streamed
</code></pre></div></div>

<p>Note the pipe between the two applications!</p>

<p><em>What new parameters did we use here?</em></p>

<p><em>What parameter is SeqScreener using that specifies how reads are input?</em></p>

<p><em>Lets look at the file SampleAC1.streamed.log?</em></p>

<div class="output">{ "hts_Stats_61829": {
    "Notes": "",
    "totalFragmentsInput": 100000,
    "totalFragmentsOutput": 100000,
    "R1_readlength_histogram": [ [151,100000] ],
    "R2_readlength_histogram": [ [151,100000] ],
    "Base_composition": {
        "A": 7410244,
        "C": 7646646,
        "G": 7854214,
        "T": 7286488,
        "N": 2408
    },
    "Single_end": {
        "SE_in": 0,
        "SE_out": 0,
        "SE_bpLen": 0,
        "SE_bQ30": 0
    },
    "Paired_end": {
        "PE_in": 100000,
        "PE_out": 100000,
        "R1_bpLen": 15100000,
        "R1_bQ30": 14481157,
        "R2_bpLen": 15100000,
        "R2_bQ30": 13852279
    }
  }, "hts_SeqScreener_61830": {
    "Notes": "",
    "totalFragmentsInput": 100000,
    "totalFragmentsOutput": 100000,
    "inverse": 0,
    "record": 1,
    "Screen_info": {
        "screenBP": 107854,
        "lookupKmers": 16765
    },
    "Single_end": {
        "SE_in": 0,
        "SE_out": 0,
        "SE_hits": 0
    },
    "Paired_end": {
        "PE_in": 100000,
        "PE_out": 100000,
        "PE_hits": 3221
    }
  }
}
</div>

<p><strong>4.</strong> A RNAseq preprocessing pipeline</p>

<ol>
  <li>hts_Stats: get stats on raw reads</li>
  <li>hts_SeqScreener: screen out (remove) phiX</li>
  <li>hts_SeqScreener: screen for (count) rrnra</li>
  <li>hts_SuperDeduper: identify and remove PCR duplicates</li>
  <li>hts_AdapterTrimmer: identify and remove adapter sequence</li>
  <li>hts_QWindowTrim: remove poor quality sequence</li>
  <li>hts_NTrimmer: remove any remaining N characters</li>
  <li>hts_CutTrim: use to remove all reads &lt; 50bp</li>
  <li>hts_Stats: get stats out output reads</li>
</ol>

<p><strong>4a.</strong> Why screen for phiX?</p>

<p>PhiX is a common control in Illumina runs, and facilities may not tell you if/when PhiX has been spiked in since it does not have a barcode, so in theory should not be in your data.</p>

<p>However:</p>
<ul>
  <li>When I know PhiX has been spiked in, I find sequence every time
    <ul>
      <li>[update] When dual matched barcodes are used, then near zero phiX reads identified.</li>
    </ul>
  </li>
  <li>When I know PhiX has not been spiked in, I do not find sequence</li>
</ul>

<p>For RNAseq and variant analysis (any mapping based technique) it is not critical to remove, but for sequence assembly it is. I think its better safe than sorry and screen for it every time.</p>

<p><strong>4b.</strong> Removing PCR duplicates with hts_SuperDeduper.</p>

<p>Removing PCR duplicates can be <strong>controversial</strong> for RNAseq, but I’m in favor of it. It tells you alot about the original complexity of each sample and potential impact of sequencing depth.</p>

<p><strong>**However, I would never do PCR duplicate removal on single-end reads!**</strong></p>

<p><img src="preproc_figures/SD_eval.png" alt="SD_eval" width="800px" /></p>

<p>Super Deduper only uses a small portion of the reads to identify duplicates.</p>

<p><img src="preproc_figures/SD_performance.png" alt="SD_performance" width="800px" /></p>

<p>We calculated the Youden Index for every combination tested and the point that acquired the highest index value (as compared to Picard MarkDuplicates) occurred at a start position at basepair 5 and a length of 10bp (20bp total over both reads). Though defaults in hts_SuperDeduper are start position at basepair 10 and a length of 10bp.</p>

<p><strong>4c.</strong> Adapter trimming by overlapping reads.</p>

<p>Consider the three scenarios below</p>

<p><strong>Insert size &gt; length of the number of cycles</strong></p>

<p><img src="preproc_figures/overlap_pairs.png" alt="overlap_pairs" width="800px" /></p>

<p>hts_AdapterTrimmer product: original pairs</p>

<p>hts_Overlapper product: original pairs</p>

<p><strong>Insert size &lt; length of the number of cycles (10bp min)</strong></p>

<p><img src="preproc_figures/overlap_single.png" alt="overlap_single" width="800px" /></p>

<p>hts_AdapterTrimmer product: original pairs</p>

<p>hts_Overlapper product: extended, single</p>

<p><strong>Insert size &lt; length of the read length</strong></p>

<p><img src="preproc_figures/overlap_adapter.png" alt="overlap_adapter" width="800px" /></p>

<p>hts_AdapterTrimmer product: adapter trimmed, pairs</p>

<p>hts_Overlapper product: adapter trimmed, single</p>

<p>Both hts_AdapterTrimmer and hts_Overlapper employ this principle to identify and remove adapters for paired-end reads. For paired-end reads the difference between the two are the output, as overlapper produces single-end reads when the pairs overlap and adapter trimmer keeps the paired end format. For single-end reads, adapter trimmer identifies and removes adapters by looking for the adapter sequence, where overlapper just ignores single-end reads (nothing to overlap).</p>

<p><strong>4d.</strong> Q-window trimming.</p>

<p>As a sequencing run progresses the quality scores tend to get worse, so its common to trim of the worst quality bases.</p>

<p><img src="preproc_figures/Qwindowtrim.png" alt="Qwindowtrim" width="800px" /></p>

<p>This is how reads commonly look, the start at “good” quality, increase to “excellent” and degrade to “poor”, with R2 always looking worse (except when they don’t) than R1 and get worse as the number of cycles increases.</p>

<p>hts_QWindowTrim trim 5’ and/or 3’ end of the sequence using a windowing (average quality in window) approach.</p>

<p><strong>4e.</strong> what does all this preprocessing get you</p>

<p>Comparing star mapping with raw and preprocessed reads</p>

<p><img src="preproc_figures/final.png" alt="final" width="400px" /></p>

<p><strong>5.</strong> Lets put it all together</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hts_Stats -L SampleAC1_htsStats.log -1 SampleAC1.subset_R1.fastq.gz -2 SampleAC1.subset_R2.fastq.gz | \
hts_SeqScreener -A -L SampleAC1_htsStats.log | \
hts_SeqScreener -s ../References/human_rrna.fasta -r -A -L SampleAC1_htsStats.log | \
hts_SuperDeduper -e 25000 -A -L SampleAC1_htsStats.log | \
hts_AdapterTrimmer -n -A -L SampleAC1_htsStats.log | \
hts_QWindowTrim -n -A -L SampleAC1_htsStats.log | \
hts_NTrimmer -n -A -L SampleAC1_htsStats.log | \
hts_CutTrim -n -m 50 -A -L SampleAC1_htsStats.log | \
hts_Stats -A -L SampleAC1_htsStats.log -f SampleAC1.htstream
</code></pre></div></div>

<p>Note the patterns:</p>
<ul>
  <li>In the first routine we use -1 and -2 to specify the original reads.</li>
  <li>In the final routine -f fastq prefix to write out new preprocessed reads.</li>
  <li>For the log, we specify -L with the same log file name for all routines, and use -A for the second routine onward to append log output, generating a single log file at the end.</li>
  <li>All other parameters are algorithm specific, can review using –help</li>
</ul>

<p><em>Review the final json output, how many reads do we have left?</em></p>

<p><em>Confirm that number by counting the number of reads in the final output files.</em></p>

<p><em>How many adapters did we detect, cut off?</em></p>

<p><em>How many PCR duplicates were there?</em></p>

<p><em>Anything else interesting?</em></p>

<hr />

<p><strong>6.</strong> We can now run the preprocessing routine across all samples on the real data using a SLURM script, <a href="/scripts/hts_preproc.slurm">hts_preproc.slurm</a>, that we should take a look at now.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example  # We'll run this from the main directory
wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/hts_preproc.slurm
less hts_preproc.slurm
</code></pre></div></div>

<p>When you are done, type “q” to exit.</p>

<div class="script">#!/bin/bash
#SBATCH --job-name=htstream # Job name
#SBATCH --nodes=1
#SBATCH --ntasks=9
#SBATCH --time=60
#SBATCH --mem=3000 # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --partition=production
#SBATCH --reservation=workshop
#SBATCH --account=workshop
#SBATCH --array=1-16
#SBATCH --output=slurmout/htstream_%A_%a.out # File to which STDOUT will be written
#SBATCH --error=slurmout/htstream_%A_%a.err # File to which STDERR will be written

start=`date +%s`
echo $HOSTNAME
echo "My SLURM_ARRAY_TASK_ID: " $SLURM_ARRAY_TASK_ID

sample=`sed "${SLURM_ARRAY_TASK_ID}q;d" samples.txt`

outpath='01-HTS_Preproc'
[[ -d ${outpath} ]] || mkdir ${outpath}
[[ -d ${outpath}/${sample} ]] || mkdir ${outpath}/${sample}

echo "SAMPLE: ${sample}"

module load htstream/1.1.0

call="hts_Stats -L ${outpath}/${sample}/${sample}_htsStats.log -1 00-RawData/${sample}/*R1* -2 00-RawData/${sample}/*R2* | \
      hts_SeqScreener -A -L ${outpath}/${sample}/${sample}_htsStats.log | \
      hts_SeqScreener -s References/human_rrna.fasta -r -A -L ${outpath}/${sample}/${sample}_htsStats.log | \
      hts_SuperDeduper -e 250000 -A -L ${outpath}/${sample}/${sample}_htsStats.log | \
      hts_AdapterTrimmer -n -A -L ${outpath}/${sample}/${sample}_htsStats.log | \
      hts_QWindowTrim -n -A -L ${outpath}/${sample}/${sample}_htsStats.log | \
      hts_NTrimmer -n -A -L ${outpath}/${sample}/${sample}_htsStats.log | \
      hts_CutTrim -n -m 50 -A -L ${outpath}/${sample}/${sample}_htsStats.log | \
      hts_Stats -A -L ${outpath}/${sample}/${sample}_htsStats.log -f ${outpath}/${sample}/${sample}"

echo $call
eval $call

end=`date +%s`
runtime=$((end-start))
echo $runtime
</div>

<p>After looking at the script, lets make a slurmout directory for the output to go and let’s run it.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir slurmout
sbatch hts_preproc.slurm  # moment of truth!
</code></pre></div></div>

<p>We can watch the progress of our task array using the ‘squeue’ command. Takes about 30 minutes to process each sample.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>squeue -u msettles  # use your username
</code></pre></div></div>

<hr />

<h2 id="quality-assurance---preprocessing-statistics-as-qaqc">Quality Assurance - Preprocessing statistics as QA/QC.</h2>

<p>Beyond generating “better” data for downstream analysis, cleaning statistics also give you an idea as to the original quality and complexity of the sample, library generation, and sequencing quality.</p>

<p>This can help inform you of how you might change your procedures in the future, either sample preparation, or in library preparation.</p>

<p>I’ve found it best to perform QA/QC on both the run as a whole (poor samples can affect other samples) and on the samples themselves as they compare to other samples (BE CONSISTENT).</p>

<p>Reports such as Basespace for Illumina, are great ways to evaluate the run as a whole, the sequencing provider usually does this for you.<br />
PCA/MDS plots of the preprocessing summary are a great way to look for technical bias across your experiment. Poor quality samples often appear as outliers on the MDS plot and can ethically be removed due to identified technical issues.</p>

<p><strong>1.</strong> Let’s make sure that all jobs completed successfully.</p>

<p>Lets first check all the “htstream_%*.out” and “htstream_%*.err” files:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example
cat slurmout/htstream_*.out
</code></pre></div></div>

<p>Look through the output and make sure you don’t see any errors. Now do the same for the err files:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cat slurmout/htstream_*.err
</code></pre></div></div>

<p>Also, check the output files. First check the number of forward and reverse output files (should be 16 each):</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd 01-HTS_Preproc
ls */*R1* | wc -l
ls */*R2* | wc -l
</code></pre></div></div>

<p>Check the sizes of the files as well. Make sure there are no zero or near-zero size files and also make sure that the size of the files are in the same ballpark as each other:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ls -lh *
</code></pre></div></div>

<p>If, for some reason, your jobs did not finish or something else went wrong, please let one of us know and we will help.</p>

<p><strong>IF</strong> for some reason it didn’t finish, is corrupted or you missed the session, you can copy over a completed copy</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cp -r /share/biocore/workshops/2019_August_RNAseq/HTS_testing /share/workshop/$USER/rnaseq_example/.
cp -r /share/biocore/workshops/2019_August_RNAseq/01-HTS_Preproc /share/workshop/$USER/rnaseq_example/.
cp  /share/biocore/workshops/2019_August_RNAseq/summary_hts.txt /share/workshop/$USER/rnaseq_example/.
</code></pre></div></div>

<hr />

<p><strong>2.</strong> Let’s take a look at the differences between the input and output files. First look at the input file:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example
zless 00-RawData/SampleAC1/SampleAC1_L3_R1.fastq.gz
</code></pre></div></div>

<p>Let’s search for the adapter sequence. Type ‘/’ (a forward slash), and then type <strong>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</strong> (the first part of the forward adapter). Press Enter. This will search for the sequence in the file and highlight each time it is found. You can now type “n” to cycle through the places where it is found. When you are done, type “q” to exit.</p>

<p>Now look at the output file:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>zless 01-HTS_Preproc/SampleAC1/SampleAC1_R1.fastq.gz
</code></pre></div></div>

<p>If you scroll through the data (using the spacebar), you will see that some of the sequences have been trimmed. Now, try searching for <strong>AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC</strong> again. You shouldn’t find it (adapters were trimmed remember), but rarely is anything perfect. You may need to use Control-C to get out of the search and then “q” to exit the ‘less’ screen.</p>

<p>Lets grep for the sequence and count occurrences</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>zcat  00-RawData/SampleAC1/SampleAC1_L3_R1.fastq.gz | grep  AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC | wc -l
zcat  01-HTS_Preproc/SampleAC1/SampleAC1_R1.fastq.gz | grep  AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC | wc -l
</code></pre></div></div>

<p><em>What is the reduction in adapters found?</em></p>

<hr />

<p><strong>3.</strong> QA/QC Summary table of the json files.</p>

<p>I’ve created a small R script to read in each json file, pull out some relevant stats and write out a table for all samples.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example  # We'll run this from the main directory
wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/summarize_stats.R

module load R
R CMD BATCH summarize_stats.R
cat summary_hts.txt
</code></pre></div></div>

<p><strong>4.</strong> Transfer summarize_stats.txt to your computer using scp or winSCP, or copy/paste from cat [sometimes doesn’t work],</p>

<p>For scp try, In a new shell session on your laptop. <strong>NOT logged into tadpole</strong>.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir ~/rnaseq_workshop
cd ~/rnaseq_workshop
scp your_username@tadpole.genomecenter.ucdavis.edu:/share/workshop/your_username/rnaseq_example/summary_hts.txt .
</code></pre></div></div>

<p>Open in excel (or excel like application), you may have to move the header column 1 cell to the right, and lets review.</p>

<p><em>Any problematic samples?</em></p>

<p><em>Anything else worth discussing?</em></p>

<h2 id="scripts">Scripts</h2>

<p>slurm script for preprocessing using slurm task array and htstream</p>

<p><a href="/scripts/hts_preproc.slurm">hts_preproc.slurm</a></p>

<p>shell script for preprocessing using bash loop and htstream.</p>

<p><a href="/scripts/hts_preproc.sh">hts_preproc.sh</a></p>

<p>R script to produce summary table, assumes exact htstream operations and order as described above.</p>

<p><a href="/scripts/summarize_stats.R">summary_stats.R</a></p>


</div>



<script src="/assets/js/scale.fix.js"></script>

<script>
var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    document.getElementById("mySidenav").style.overflowY = "hidden";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    document.getElementById("mySidenav").style.overflowY = "visible";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};

// store the sidenav scroll pos for loading or set the pos
sn = document.getElementById("mySidenav");
sn.onscroll = function() {sessionStorage.setItem("scrollpos",sn.scrollTop);};
window.onload = function() {sn.scrollTop = sessionStorage.getItem("scrollpos");};

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
