
<link rel="stylesheet" href="/assets/highlight/styles/default.css">
<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;margin:5;background-color:#CFB53B; cursor:pointer;padding-right:50px">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="padding-left:50px;padding-top:10px;background-color:#CFB53B;">UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a>

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>

<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:160px">

<h1 id="alignment-using-salmon">Alignment using Salmon</h1>

<h2 id="salmon-aligner">Salmon Aligner</h2>
<p><a href="https://salmon.readthedocs.io/en/latest/salmon.html">Salmon</a> is a tool for quantifying the expression of transcripts using RNA-seq data. Salmon uses new algorithms (specifically, coupling the concept of quasi-mapping with a two-phase inference procedure) to provide accurate expression estimates very quickly (i.e. wicked-fast) and while using little memory. Salmon performs its inference using an expressive and realistic model of RNA-seq data that takes into account experimental attributes and biases commonly observed in real RNA-seq data.</p>

<h2 id="we-first-need-to-index-the-reference-in-this-case-the-transcriptome">We first need to index the reference (In this case the transcriptome)</h2>

<p><strong>1.</strong> First lets make sure we are where we are supposed to be and that the References directory is available.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example
</code></pre></div></div>

<hr />
<p><strong>2.</strong> To align our data we will need the transcriptome (fasta) and annotation (gtf) for human. There are many places to find them, but we are going to get it from the <a href="https://www.gencodegenes.org/human/">GENCODE</a>.</p>

<p>We need to first get the urls for the and protein coding genes. For RNAseq we want to use the protein coding transcript sequences and basic gene annotation. At the time of this workshop the current version of GENCODE is <em>31</em>. You will want to update the scripts to use the current version.</p>

<p><img src="alignment_figures/index_figure1.png" alt="index_figure1" width="800px" /></p>

<p><img src="alignment_figures/index_figure3.png" alt="index_figure1" width="800px" /></p>

<hr />
<p><strong>3.</strong> Lets take a look at the help docs for salmon and its subcommands as well:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load salmon
salamon -h
salmon index -h
</code></pre></div></div>

<hr />
<p><strong>4.</strong> First we need to index the transcriptome for STAR. Lets pull down a slurm script to index the human GENCODE version of the transcriptome.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/salmon_index.slurm
less salmon_index.slurm
</code></pre></div></div>

<p>Press ‘q’ to exit.</p>

<div class="script">#!/bin/bash

#SBATCH --job-name=salmon_index # Job name
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --time=60
#SBATCH --mem=15000 # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --partition=production
#SBATCH --reservation=workshop
#SBATCH --account=workshop
#SBATCH --output=slurmout/salmon-index_%A.out # File to which STDOUT will be written
#SBATCH --error=slurmout/salmon-index_%A.err # File to which STDERR will be written

start=`date +%s`
echo $HOSTNAME

outpath="References"

cd ${outpath}
wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_31/gencode.v31.pc_transcripts.fa.gz
gunzip gencode.v31.pc_transcripts.fa.gz
PC_FASTA="gencode.v31.pc_transcripts.fa"
INDEX="salmon_gencode.v31.index"

module load salmon
call="salmon index -i ${INDEX} -k 31 --gencode -p 8 -t ${PC_FASTA}"
echo $call
eval $call

end=`date +%s`
runtime=$((end-start))
echo $runtime
</div>

<ol>
  <li>The script changes into the References directory.</li>
  <li>It uses wget to download the transcript fasta file from GENCODE.</li>
  <li>Uncompresses it using gunzip.</li>
  <li>Run Salmon indexing, using the “gencode” flag to parse the GENCODE file properly, and outputting to a new directory called “salmon_gencode.v31.index”.</li>
</ol>

<p>Run salmon indexing when ready.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sbatch salmon_index.slurm
</code></pre></div></div>

<p>This step does not take long, about 15 minutes. You can look at the <a href="https://salmon.readthedocs.io/en/latest/salmon.html">salmon documentation</a> while you wait. All of the output files will be written to the salmon_gencode.v31.index directory.</p>

<p><strong>IF</strong> for some reason it didn’t finish, is corrupted, or you missed the session, you can copy over a completed copy.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cp -r /share/biocore/workshops/2019_August_RNAseq/References/salmon_gencode.v31.index /share/workshop/$USER/rnaseq_example/References/
</code></pre></div></div>

<h2 id="alignments">Alignments</h2>

<p><strong>1.</strong> We are now ready to try an alignment:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example/HTS_testing
</code></pre></div></div>

<p>Then run the salmon quant (quantify transcripts) command</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load salmon
	salmon quant --help-reads

salmon quant \
--threads 8 \
--index ../References/salmon_gencode.v31.index \
	--libType A \
	--validateMappings \
	--geneMap ../References/gencode.v31.primary_assembly.annotation.gtf \
	--output SampleAC1.salmon \
-1 SampleAC1.streamed_R1.fastq.gz \
	-2 SampleAC1.streamed_R2.fastq.gz
</code></pre></div></div>

<p>In the command, we are telling salmon to quantify reads with libtype ‘auto’ (<a href="https://salmon.readthedocs.io/en/latest/salmon.html#what-s-this-libtype">libtype</a>) on a gene level (‘–geneMap’), the folder for all the output files will be SampleAC1.salmon, and finally, the input file pair.</p>

<h2 id="running-salmon-on-the-experiment">Running Salmon on the experiment</h2>

<p><strong>1.</strong> We can now run Salmon across all samples on the real data using a SLURM script, <a href="/scripts/salmon.slurm">salmon.slurm</a>, that we should take a look at now.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example  # We'll run this from the main directory
wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/salmon.slurm
less salmon.slurm
</code></pre></div></div>

<p>When you are done, type “q” to exit.</p>

<div class="script">#!/bin/bash

#SBATCH --array=1-16
#SBATCH --job-name=salmon # Job name
#SBATCH --nodes=1
#SBATCH --ntasks=8
#SBATCH --time=1440
#SBATCH --mem=20000 # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --partition=production
#SBATCH --reservation=workshop
#SBATCH --account=workshop
#SBATCH --output=slurmout/salmon_%A_%a.out # File to which STDOUT will be written
#SBATCH --error=slurmout/salmon_%A_%a.err # File to which STDERR will be written


start=`date +%s`
hostname

outdir="02-Salmon_alignment"
sampfile="samples.txt"
REF="References/salmon_gencode.v31.index"
GTF="References/gencode.v31.primary_assembly.annotation.gtf"

SAMPLE=`head -n ${SLURM_ARRAY_TASK_ID} $sampfile | tail -1`
R1="01-HTS_Preproc/$SAMPLE/${SAMPLE}_R1.fastq.gz"
R2="01-HTS_Preproc/$SAMPLE/${SAMPLE}_R2.fastq.gz"

echo $SAMPLE

if [ ! -e $outdir ]; then
    mkdir $outdir
fi

module load salmon

call="salmon quant \
      --threads 8 \
      --index ${REF} \
      --libType A \
      --validateMappings \
      --geneMap ${GTF} \
      --output $outdir/$SAMPLE \
      -1 $R1 \
      -2 $R2"

echo $call
eval $call

end=`date +%s`
runtime=$((end-start))
echo Runtime: $runtime seconds
</div>

<ol>
  <li>Loads the salmon module and then runs salmon in “quant” mode to quantify (i.e. count) the reads aligning to transcripts. Salmon uses the annotation GTF file to roll up the transcript counts into gene-level counts.</li>
  <li>The script specifies the output directory (02-Salmon_alignment), the samples file (samples.txt), the reference that we just indexed, and the annotation that we downloaded.</li>
  <li>Creates the output directory.</li>
  <li>It then defines the filenames for the forward and reverse reads (R1 and R2).</li>
</ol>

<p><strong>2.</strong> After looking at the script, lets run it.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sbatch salmon.slurm  # moment of truth!
</code></pre></div></div>

<p>We can watch the progress of our task array using the ‘squeue’ command. Takes about 30 minutes to process each sample.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>squeue -u msettles  # use your username
</code></pre></div></div>

<p><strong>3.</strong> Once the jobs finish, take a look at one of the output files:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>	cd /share/workshop/$USER/rnaseq_example/02-Salmon_alignment/SampleAC1
	head quant.sf
</code></pre></div></div>

<div class="script">Name	Length	EffectiveLength	TPM	NumReads
ENST00000641515.2	2618	2404.977	0.000000	0.000
ENST00000335137.4	1054	840.977	0.000000	0.000
ENST00000426406.3	995	781.997	1.678844	2.000
ENST00000420190.6	1578	1364.977	0.000000	0.000
ENST00000437963.5	387	186.118	0.000000	0.000
ENST00000342066.8	2557	2343.977	0.000000	0.000
ENST00000618181.4	2179	1965.977	0.000000	0.000
ENST00000622503.4	2557	2343.977	0.000000	0.000
ENST00000618323.4	2159	1945.977	0.000000	0.000
</div>

<p>These are the transcript-level counts. Each row describes a single quantification record. The columns have the following interpretation.</p>

<ul>
  <li>Name — This is the name of the target transcript provided in the input transcript database (FASTA file).</li>
  <li>Length — This is the length of the target transcript in nucleotides.</li>
  <li>EffectiveLength — This is the computed effective length of the target transcript. It takes into account all factors being modeled that will effect the probability of sampling fragments from this transcript, including the fragment length distribution and sequence-specific and gc-fragment bias (if they are being modeled).</li>
  <li>TPM — This is salmon’s estimate of the relative abundance of this transcript in units of Transcripts Per Million (TPM). TPM is the recommended relative abundance measure to use for downstream analysis.</li>
  <li>NumReads — This is salmon’s estimate of the number of reads mapping to each transcript that was quantified. It is an “estimate” insofar as it is the expected number of reads that have originated from each transcript given the structure of the uniquely mapping and multi-mapping reads and the relative abundance estimates for each transcript.</li>
</ul>

<p>Gene level quantification can be found in the quant.genes.sf file.</p>

<h2 id="quality-assurance---mapping-statistics-as-qaqc">Quality Assurance - Mapping statistics as QA/QC.</h2>

<p><strong>1.</strong> Once your jobs have finished successfully (check the error and out logs like we did in the previous exercise), use a script of ours, <a href="../scripts/salmon_stats.sh">salmon_stats.sh</a> to collect the alignment stats. Don’t worry about the script’s contents at the moment; you’ll use very similar commands to create a counts table in the next section. For now:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example  # We'll run this from the main directory
wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/scripts/salmon_stats.sh
	module load R
	R CMD BATCH salmon_stats.R
</code></pre></div></div>

<p><strong>2.</strong> Transfer summary_salmon_alignments.txt to your computer using scp or winSCP, or copy/paste from cat [sometimes doesn’t work],</p>

<p>In a new shell session on your laptop. <strong>NOT logged into tadpole</strong>.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir ~/rnaseq_workshop
cd ~/rnaseq_workshop
scp [your_username]@tadpole.genomecenter.ucdavis.edu:/share/workshop/[your_username]/rnaseq_example/summary_alignments.txt .
</code></pre></div></div>

<p>Its ok of the mkdir command fails (“File exists”) because we aleady created the directory earlier.</p>

<p>Open in excel (or excel like application), and lets review.</p>

<p>The table that this script creates (“alignment_salmon_stats.txt”) can be pulled to your laptop via ‘scp’, or WinSCP, etc., and imported into a spreadsheet. Are all samples behaving similarly? Discuss …</p>

<h2 id="generating-gene-level-read-counts-table">Generating gene level read counts table</h2>

<p><strong>1.</strong> Similar to STAR, we want to do these steps for ALL of the read count files… and to do that we will be using a ‘for loop’ directly on the command line. First, just run a simple ‘for loop’ that will print out the names of all the files we want to use:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>for sample in `cat samples.txt`; do echo ${sample}; done
</code></pre></div></div>

<p>This command takes all the files that we listed in samples.txt and loops through them, one by one, and for every iteration, assigns the filename to the ‘${sample}’ variable. Also, for every iteration, it runs whatever commands are between the ‘do’ and ‘done’…. and every iteration the value of ‘${sample}’ changes. The semi-colons separate the parts of the loop. The ‘echo’ command just prints the value of $x to the screen… in this case just the filename. For salmon we want to pull the TPM column (column 4) as the best estimate for transcript abundance:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER/rnaseq_example
mkdir 03-Counts
mkdir 03-Counts/tmp
for sample in `cat samples.txt`; do \
    echo ${sample}
    cat 02-Salmon_alignment/${sample}/quant.genes.sf | cut -f4 &gt; 03-Counts/tmp/${sample}.count
done
</code></pre></div></div>

<p>After this command, there should be a counts file for every sample, in 03-Counts/tmp.</p>

<hr />
<p><strong>4.</strong> Next, we need to get the columns for the final table. Because all of these files are sorted in the exact same order (by gene ID), we can just use the columns from any of the files:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>tail 02-Salmon_alignment/SampleAC1/quant.genes.sf | cut -f1 &gt; 03-Counts/tmp/geneids.txt
head 03-Counts/tmp/geneids.txt
</code></pre></div></div>

<p>Finally, we want to combine all of these columns together using the ‘paste’ command, and put it in a temporary file:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>paste 03-Counts/tmp/geneids.txt 03-Counts/tmp/*.count &gt; 03-Counts/tmp/tmp.out
</code></pre></div></div>

<hr />
<p><strong>5.</strong> The final step is to create a header of sample names and combine it with the temp file. The header is just all of the sample names separated by tabs. And again, since we pasted the columns in sorted order (wildcards automatically sort in order), the columns just need to be in that same order.</p>

<p>We take the samples.txt file and pipe that to the sort (to ensure they are in the same order) and then ‘paste’ command with the ‘-s’ option, which takes a column of values and transposes them into a row, separated by the tab character. And finally, let’s put everything together:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cat &lt;(cat samples.txt | sort | paste -s) 03-Counts/tmp/tmp.out &gt; 03-Counts/rnaseq_salmon_workshop_counts.txt
rm -rf 03-Counts/tmp
head 03-Counts/rnaseq_salmon_workshop_counts.txt
</code></pre></div></div>

<h2 id="scripts">Scripts</h2>

<p>slurm script for indexing the genome</p>

<p><a href="/scripts/salmon_index.slurm">salmon_index.slurm</a></p>

<p>shell script for indexing the genome</p>

<p><a href="/scripts/salmon_index.sh">salmon_index.sh</a></p>

<p>slurm script for mapping using slurm task array and star</p>

<p><a href="/scripts/salmon.slurm">salmon.slurm</a></p>

<p>shell script for mapping using bash loop and star.</p>

<p><a href="/scripts/salmon.sh">salmon.sh</a></p>

<p>R script to produce summary mapping table</p>

<p><a href="/scripts/salmon_stats.R">salmon_stats.R</a></p>


</div>



<script src="/assets/js/scale.fix.js"></script>



<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'true', 'auto');
    ga('send', 'pageview');
</script>


<script>


var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};
//var scrollpos=0;
//document.getElementById("mySidenav").onscroll = function() {scrollpos = document.getElementById("mySidenav").scrollTop;}

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
