
<link rel="stylesheet" href="/assets/highlight/styles/default.css">
<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;margin:5;background-color:#CFB53B; cursor:pointer;padding-right:50px">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="padding-left:50px;padding-top:10px;background-color:#CFB53B;">UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a>

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>

<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:160px">

<h3 id="create-a-new-rstudio-project">Create a new RStudio project</h3>

<p>Open RStudio and create a new project, for more info see <a href="https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects">https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects</a></p>

<ul>
  <li>File &gt; New Project &gt; New Directory &gt; New Project (name the new directory, Ex. Differential_Expression) and check “use packrat with this project” if present.</li>
</ul>

<p>Learn more about packrat see <a href="https://rstudio.github.io/packrat/">https://rstudio.github.io/packrat/</a></p>

<p>Set some options and make sure the packages edgeR, gplots, RColorBrewer, topGO, KEGGREST, Rgraphviz and org.Hs.eg.db are installed (if not install it), and then load</p>

<p>In the R console run the following commands</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w">
</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">any</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">installed.packages</span><span class="p">())</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"edgeR"</span><span class="p">)){</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w">
      </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w">
  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"edgeR"</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span><span class="w">

</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">any</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">installed.packages</span><span class="p">())</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"topGO"</span><span class="p">)){</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w">
      </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w">
  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"topGO"</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">topGO</span><span class="p">)</span><span class="w">

</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">any</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">installed.packages</span><span class="p">())</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"KEGGREST"</span><span class="p">)){</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w">
      </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w">
  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"KEGGREST"</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">KEGGREST</span><span class="p">)</span><span class="w">

</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">any</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">installed.packages</span><span class="p">())</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"Rgraphviz"</span><span class="p">)){</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w">
      </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w">
  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"Rgraphviz"</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">Rgraphviz</span><span class="p">)</span><span class="w">

</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">any</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">installed.packages</span><span class="p">())</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"org.Hs.eg.db"</span><span class="p">)){</span><span class="w">
  </span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="n">requireNamespace</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">,</span><span class="w"> </span><span class="n">quietly</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w">
      </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"BiocManager"</span><span class="p">)</span><span class="w">
  </span><span class="n">BiocManager</span><span class="o">::</span><span class="n">install</span><span class="p">(</span><span class="s2">"org.Hs.eg.db"</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span><span class="w">

</span><span class="n">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span><span class="w">
</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">any</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">installed.packages</span><span class="p">())</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"gplots"</span><span class="p">)){</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"gplots"</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">gplots</span><span class="p">)</span><span class="w">

</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">any</span><span class="p">(</span><span class="n">rownames</span><span class="p">(</span><span class="n">installed.packages</span><span class="p">())</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"RColorBrewer"</span><span class="p">)){</span><span class="w">
</span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"RColorBrewer"</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">RColorBrewer</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h3 id="download-the-template-markdown-workshop-document-and-open-it">Download the template Markdown workshop document and open it</h3>

<p>In the R console run the following command</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">download.file</span><span class="p">(</span><span class="s2">"https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/differential_expression/DE_Analysis.Rmd"</span><span class="p">,</span><span class="w"> </span><span class="s2">"DE_Analysis.Rmd"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h3 id="download-the-data-file-for-the-workshop-document-and-previewopen-it">Download the data file for the workshop document and preview/open it</h3>

<p>This is the the counts file generated after running <a href="https://ucdavis-bioinformatics-training.github.io/2019_August_UCD_mRNAseq_Workshop/data_reduction/counts.html">Generating Summarized Counts</a>. I’ve also uploaded to the github pages</p>

<p>In the R console run the following command.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">download.file</span><span class="p">(</span><span class="s2">"https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/differential_expression/de_data/rnaseq_workshop_counts.txt"</span><span class="p">,</span><span class="w"> </span><span class="s2">"rnaseq_workshop_counts.txt"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<h3 id="edit-the-file-yaml-portion">Edit the file YAML portion</h3>

<p>The top YAML (YAML ain’t markup language) portion of the doc tells RStudio how to parse the document.</p>

<pre><code>---
title: "Data_in_R"
author: your_name
date: current_date
output:
    html_notebook: default
    html_document: default
---</code></pre>


</div>



<script src="/assets/js/scale.fix.js"></script>



<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'true', 'auto');
    ga('send', 'pageview');
</script>


<script>


var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};
//var scrollpos=0;
//document.getElementById("mySidenav").onscroll = function() {scrollpos = document.getElementById("mySidenav").scrollTop;}

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
