
<link rel="stylesheet" href="/assets/highlight/styles/default.css">
<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;margin:5;background-color:#CFB53B; cursor:pointer;padding-right:50px">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="padding-left:50px;padding-top:10px;background-color:#CFB53B;">UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a>

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>

<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:160px">

<h1 id="go-and-kegg-enrichment-analysis">GO AND KEGG Enrichment Analysis</h1>

<p>Load libraries</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">topGO</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: BiocGenerics
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: parallel
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Attaching package: 'BiocGenerics'
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply, Map,
##     mapply, match, mget, order, paste, pmax, pmax.int, pmin,
##     pmin.int, Position, rank, rbind, Reduce, rownames, sapply,
##     setdiff, sort, table, tapply, union, unique, unsplit, which,
##     which.max, which.min
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: graph
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: Biobase
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: GO.db
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: AnnotationDbi
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: stats4
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: IRanges
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: S4Vectors
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Attaching package: 'S4Vectors'
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following object is masked from 'package:base':
## 
##     expand.grid
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: SparseM
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Attaching package: 'SparseM'
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following object is masked from 'package:base':
## 
##     backsolve
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## groupGOTerms: 	GOBPTerm, GOMFTerm, GOCCTerm environments built.
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Attaching package: 'topGO'
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following object is masked from 'package:IRanges':
## 
##     members
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">KEGGREST</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">org.Hs.eg.db</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
</code></pre></div></div>

<p>Files for examples created in the DE analysis</p>

<h2 id="gene-ontology-go-enrichment">Gene Ontology (GO) Enrichment</h2>

<p>Gene ontology (http://www.geneontology.org/) provides a controlled vocabulary for describing biological processes (BP ontology), molecular functions (MF ontology) and cellular components (CC ontology)</p>

<p>The GO ontologies themselves are organism-independent; terms are associated with genes for a specific organism through direct experimentation or through sequence homology with another organism and its GO annotation.</p>

<p>Terms are related to other terms through parent-child relationships in a directed acylic graph.</p>

<p>Enrichment analysis provides one way of drawing conclusions about a set of differential expression results.</p>

<p><strong>1.</strong> topGO Example Using Kolmogorov-Smirnov Testing
Our first example uses Kolmogorov-Smirnov Testing for enrichment testing of our human DE results, with GO annotation obtained from the Bioconductor database org.Hs.eg.db.</p>

<p>The first step in each topGO analysis is to create a topGOdata object.  This contains the genes, the score for each gene (here we use the p-value from the DE test), the GO terms associated with each gene, and the ontology to be used (here we use the biological process ontology)</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">infile</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"A.C_v_B.C.txt"</span><span class="w">
</span><span class="n">tmp</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.delim</span><span class="p">(</span><span class="n">infile</span><span class="p">)</span><span class="w">

</span><span class="n">geneList</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">tmp</span><span class="o">$</span><span class="n">P.Value</span><span class="w">
</span><span class="n">xx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.list</span><span class="p">(</span><span class="n">org.Hs.egENSEMBL2EG</span><span class="p">)</span><span class="w">
</span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">xx</span><span class="p">[</span><span class="n">tmp</span><span class="o">$</span><span class="n">Gene</span><span class="p">]</span><span class="w">

</span><span class="c1"># Create topGOData object</span><span class="w">
</span><span class="n">GOdata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">new</span><span class="p">(</span><span class="s2">"topGOdata"</span><span class="p">,</span><span class="w">
	</span><span class="n">ontology</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"BP"</span><span class="p">,</span><span class="w">
	</span><span class="n">allGenes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span><span class="w">
	</span><span class="n">geneSelectionFun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="n">x</span><span class="p">,</span><span class="w">
	</span><span class="n">annot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annFUN.org</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"org.Hs.eg.db"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Building most specific GOs .....
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 	( 11330 GO terms found. )
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Build GO DAG topology ..........
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 	( 15346 GO terms and 36423 relations. )
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Annotating nodes ...............
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 	( 11714 genes annotated to the GO terms. )
</code></pre></div></div>

<p><strong>2.</strong> The topGOdata object is then used as input for enrichment testing:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Kolmogorov-Smirnov testing</span><span class="w">
</span><span class="n">resultKS</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">runTest</span><span class="p">(</span><span class="n">GOdata</span><span class="p">,</span><span class="w"> </span><span class="n">algorithm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"weight01"</span><span class="p">,</span><span class="w"> </span><span class="n">statistic</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"ks"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 			 -- Weight01 Algorithm -- 
## 
## 		 the algorithm is scoring 15346 nontrivial nodes
## 		 parameters: 
## 			 test statistic: ks
## 			 score order: increasing
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 20:	1 nodes to be scored	(0 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 19:	5 nodes to be scored	(0 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 18:	21 nodes to be scored	(1 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 17:	47 nodes to be scored	(26 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 16:	118 nodes to be scored	(67 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 15:	242 nodes to be scored	(170 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 14:	475 nodes to be scored	(481 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 13:	843 nodes to be scored	(1128 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 12:	1255 nodes to be scored	(2277 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 11:	1666 nodes to be scored	(4235 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 10:	1989 nodes to be scored	(5807 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 9:	2100 nodes to be scored	(7059 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 8:	1996 nodes to be scored	(8561 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 7:	1877 nodes to be scored	(9519 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 6:	1410 nodes to be scored	(10407 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 5:	775 nodes to be scored	(10892 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 4:	367 nodes to be scored	(11258 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 3:	134 nodes to be scored	(11447 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 2:	24 nodes to be scored	(11512 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 1:	1 nodes to be scored	(11562 eliminated genes)
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">tab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GenTable</span><span class="p">(</span><span class="n">GOdata</span><span class="p">,</span><span class="w"> </span><span class="n">raw.p.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">resultKS</span><span class="p">,</span><span class="w"> </span><span class="n">topNodes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">resultKS</span><span class="o">@</span><span class="n">score</span><span class="p">),</span><span class="w"> </span><span class="n">numChar</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">120</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p>topGO preferentially tests more specific terms, utilizing the topology of the GO graph. The algorithms used are described in detail <a href="https://academic.oup.com/bioinformatics/article/22/13/1600/193669">here</a>.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">head</span><span class="p">(</span><span class="n">tab</span><span class="p">,</span><span class="w"> </span><span class="m">15</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##         GO.ID
## 1  GO:0008150
## 2  GO:0007165
## 3  GO:0045787
## 4  GO:0045104
## 5  GO:0006909
## 6  GO:0051057
## 7  GO:0042058
## 8  GO:0060088
## 9  GO:0031016
## 10 GO:2000503
## 11 GO:0008285
## 12 GO:0030177
## 13 GO:0018108
## 14 GO:0003215
## 15 GO:0048739
##                                                                Term
## 1                                                biological_process
## 2                                               signal transduction
## 3                                 positive regulation of cell cycle
## 4                   intermediate filament cytoskeleton organization
## 5                                                      phagocytosis
## 6  positive regulation of small GTPase mediated signal transduction
## 7  regulation of epidermal growth factor receptor signaling pathway
## 8                  auditory receptor cell stereocilium organization
## 9                                              pancreas development
## 10            positive regulation of natural killer cell chemotaxis
## 11                        negative regulation of cell proliferation
## 12                     positive regulation of Wnt signaling pathway
## 13                                peptidyl-tyrosine phosphorylation
## 14                            cardiac right ventricle morphogenesis
## 15                                 cardiac muscle fiber development
##    Annotated Significant Expected raw.p.value
## 1      11714       11714    11714     &lt; 1e-30
## 2       4310        4310     4310     0.00030
## 3        313         313      313     0.00058
## 4         35          35       35     0.00148
## 5        220         220      220     0.00161
## 6         51          51       51     0.00166
## 7         69          69       69     0.00212
## 8          8           8        8     0.00274
## 9         68          68       68     0.00290
## 10         6           6        6     0.00320
## 11       600         600      600     0.00364
## 12       143         143      143     0.00402
## 13       302         302      302     0.00410
## 14        18          18       18     0.00410
## 15         9           9        9     0.00415
</code></pre></div></div>
<ul>
  <li>Annotated: number of genes (in our gene list) that are annotated with the term</li>
  <li>Significant: n/a for this example, same as Annotated here</li>
  <li>Expected: n/a for this example, same as Annotated here</li>
  <li>raw.p.value: P-value from Kolomogorov-Smirnov test that DE p-values annotated with the term are smaller (i.e. more significant) than those not annotated with the term.</li>
</ul>

<p>The Kolmogorov-Smirnov test directly compares two probability distributions based on their maximum distance.</p>

<p>To illustrate the KS test, we plot probability distributions of p-values that are and that are not annotated with the term “phosphatidylinositol phosphorylation”.  (This won’t exactly match what topGO does due to their elimination algorithm):</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">rna.pp.terms</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">genesInTerm</span><span class="p">(</span><span class="n">GOdata</span><span class="p">)[[</span><span class="s2">"GO:0046854"</span><span class="p">]]</span><span class="w"> </span><span class="c1"># get genes associated with term</span><span class="w">
</span><span class="n">p.values.in</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">geneList</span><span class="p">[</span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">rna.pp.terms</span><span class="p">]</span><span class="w">
</span><span class="n">p.values.out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">geneList</span><span class="p">[</span><span class="o">!</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">rna.pp.terms</span><span class="p">)]</span><span class="w">
</span><span class="n">plot.ecdf</span><span class="p">(</span><span class="n">p.values.in</span><span class="p">,</span><span class="w"> </span><span class="n">verticals</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w"> </span><span class="n">do.points</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"red"</span><span class="p">,</span><span class="w"> </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">xlim</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">),</span><span class="w">
          </span><span class="n">main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Empirical Distribution of DE P-Values by Annotation with 'RNA phosphatidylinositol phosphorylation'"</span><span class="p">,</span><span class="w">
          </span><span class="n">cex.main</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">,</span><span class="w"> </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"p"</span><span class="p">,</span><span class="w"> </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Probabilty(P-Value &lt; p)"</span><span class="p">)</span><span class="w">
</span><span class="n">ecdf.out</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ecdf</span><span class="p">(</span><span class="n">p.values.out</span><span class="p">)</span><span class="w">
</span><span class="n">xx</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unique</span><span class="p">(</span><span class="n">sort</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">seq</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">201</span><span class="p">),</span><span class="w"> </span><span class="n">knots</span><span class="p">(</span><span class="n">ecdf.out</span><span class="p">))))</span><span class="w">
</span><span class="n">lines</span><span class="p">(</span><span class="n">xx</span><span class="p">,</span><span class="w"> </span><span class="n">ecdf.out</span><span class="p">(</span><span class="n">xx</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w">
</span><span class="n">legend</span><span class="p">(</span><span class="s2">"bottomright"</span><span class="p">,</span><span class="w"> </span><span class="n">legend</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"Genes Annotated with 'phosphatidylinositol phosphorylation'"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Genes Not Annotated with 'phosphatidylinositol phosphorylation'"</span><span class="p">),</span><span class="w"> </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="o">:</span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.9</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><img src="/differential_expression/enrichment_files/figure-html/unnamed-chunk-5-1.png" alt="" /><!-- --></p>

<p>We can use the function showSigOfNodes to plot the GO graph for the 3 most significant terms and their parents, color coded by enrichment p-value (red is most significant):</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">par</span><span class="p">(</span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.3</span><span class="p">)</span><span class="w">
</span><span class="n">showSigOfNodes</span><span class="p">(</span><span class="n">GOdata</span><span class="p">,</span><span class="w"> </span><span class="n">score</span><span class="p">(</span><span class="n">resultKS</span><span class="p">),</span><span class="w"> </span><span class="n">firstSigNodes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">useInfo</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"def"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: Rgraphviz
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## Loading required package: grid
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Attaching package: 'grid'
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following object is masked from 'package:topGO':
## 
##     depth
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Attaching package: 'Rgraphviz'
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following objects are masked from 'package:IRanges':
## 
##     from, to
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## The following objects are masked from 'package:S4Vectors':
## 
##     from, to
</code></pre></div></div>

<p><img src="/differential_expression/enrichment_files/figure-html/unnamed-chunk-6-1.png" alt="" /><!-- --></p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## $dag
## A graphNEL graph with directed edges
## Number of Nodes = 15 
## Number of Edges = 27 
## 
## $complete.dag
## [1] "A graph with 15 nodes."
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">par</span><span class="p">(</span><span class="n">cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><strong>3.</strong> topGO Example Using Fisher’s Exact Test
Next, we use Fisher’s exact test to test for GO enrichment among significantly DE genes.</p>

<p>Create topGOdata object:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Create topGOData object</span><span class="w">
</span><span class="n">GOdata</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">new</span><span class="p">(</span><span class="s2">"topGOdata"</span><span class="p">,</span><span class="w">
	</span><span class="n">ontology</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"BP"</span><span class="p">,</span><span class="w">
	</span><span class="n">allGenes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">geneList</span><span class="p">,</span><span class="w">
	</span><span class="n">geneSelectionFun</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.05</span><span class="p">),</span><span class="w">
	</span><span class="n">annot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annFUN.org</span><span class="w"> </span><span class="p">,</span><span class="w"> </span><span class="n">mapping</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"org.Hs.eg.db"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Building most specific GOs .....
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 	( 11330 GO terms found. )
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Build GO DAG topology ..........
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 	( 15346 GO terms and 36423 relations. )
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## Annotating nodes ...............
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 	( 11714 genes annotated to the GO terms. )
</code></pre></div></div>

<p>Run Fisher’s Exact Test:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">resultFisher</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">runTest</span><span class="p">(</span><span class="n">GOdata</span><span class="p">,</span><span class="w"> </span><span class="n">algorithm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"elim"</span><span class="p">,</span><span class="w"> </span><span class="n">statistic</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"fisher"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 			 -- Elim Algorithm -- 
## 
## 		 the algorithm is scoring 12563 nontrivial nodes
## 		 parameters: 
## 			 test statistic: fisher
## 			 cutOff: 0.01
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 19:	1 nodes to be scored	(0 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 18:	15 nodes to be scored	(0 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 17:	36 nodes to be scored	(0 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 16:	82 nodes to be scored	(0 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 15:	183 nodes to be scored	(0 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 14:	370 nodes to be scored	(127 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 13:	644 nodes to be scored	(127 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 12:	954 nodes to be scored	(1939 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 11:	1311 nodes to be scored	(2378 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 10:	1585 nodes to be scored	(3443 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 9:	1730 nodes to be scored	(4570 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 8:	1668 nodes to be scored	(5374 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 7:	1571 nodes to be scored	(7408 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 6:	1229 nodes to be scored	(8680 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 5:	691 nodes to be scored	(10072 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 4:	341 nodes to be scored	(11005 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 3:	129 nodes to be scored	(11281 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 2:	22 nodes to be scored	(11405 eliminated genes)
</code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## 
## 	 Level 1:	1 nodes to be scored	(11488 eliminated genes)
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">tab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">GenTable</span><span class="p">(</span><span class="n">GOdata</span><span class="p">,</span><span class="w"> </span><span class="n">raw.p.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">resultFisher</span><span class="p">,</span><span class="w"> </span><span class="n">topNodes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">resultFisher</span><span class="o">@</span><span class="n">score</span><span class="p">),</span><span class="w">
				</span><span class="n">numChar</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">120</span><span class="p">)</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">tab</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##        GO.ID                                     Term Annotated
## 1 GO:0008150                       biological_process     11714
## 2 GO:0009987                         cellular process     10689
## 3 GO:0007165                      signal transduction      4310
## 4 GO:0043066 negative regulation of apoptotic process       755
## 5 GO:0044237               cellular metabolic process      7587
## 6 GO:0043065 positive regulation of apoptotic process       548
##   Significant Expected raw.p.value
## 1        4612  4430.19     &lt; 1e-30
## 2        4213  4042.54     1.4e-11
## 3        1756  1630.02     2.9e-07
## 4         325   285.54     3.2e-06
## 5        2973  2869.37     6.0e-06
## 6         235   207.25     2.4e-05
</code></pre></div></div>
<ul>
  <li>Annotated: number of genes (in our gene list) that are annotated with the term</li>
  <li>Significant: Number of significantly DE genes annotated with that term (i.e. genes where geneList = 1)</li>
  <li>Expected: Under random chance, number of genes that would be expected to be significantly DE and annotated with that term</li>
  <li>raw.p.value: P-value from Fisher’s Exact Test, testing for association between significance and pathway membership.</li>
</ul>

<p>Fisher’s Exact Test is applied to the table:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center"><strong>Significance/Annotation</strong></th>
      <th style="text-align: center"><strong>Annotated With GO Term</strong></th>
      <th style="text-align: center"><strong>Not Annotated With GO Term</strong></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><strong>Significantly DE</strong></td>
      <td style="text-align: center">n1</td>
      <td style="text-align: center">n3</td>
    </tr>
    <tr>
      <td style="text-align: center"><strong>Not Significantly DE</strong></td>
      <td style="text-align: center">n2</td>
      <td style="text-align: center">n4</td>
    </tr>
  </tbody>
</table>

<p>and compares the probability of the observed table, conditional on the row and column sums, to what would be expected under random chance.</p>

<p>Advantages over KS (or Wilcoxon) Tests:</p>

<p>*Ease of interpretation</p>

<p>Disadvantages:</p>

<ul>
  <li>Relies on significant/non-significant dichotomy (an interesting gene could have an adjusted p-value of 0.051 and be counted as non-significant)</li>
  <li>Less powerful</li>
  <li>May be less useful if there are very few (or a large number of) significant genes</li>
</ul>

<p>##. KEGG Pathway Enrichment Testing With KEGGREST
KEGG, the Kyoto Encyclopedia of Genes and Genomes (https://www.genome.jp/kegg/), provides assignment of genes for many organisms into pathways.</p>

<p>We will access KEGG pathway assignments for human through the KEGGREST Bioconductor package, and then use some homebrew code for enrichment testing.</p>

<p><strong>1.</strong> Get all human pathways and their genes:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Pull all pathways for AT</span><span class="w">
</span><span class="n">pathways.list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">keggList</span><span class="p">(</span><span class="s2">"pathway"</span><span class="p">,</span><span class="w"> </span><span class="s2">"hsa"</span><span class="p">)</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">pathways.list</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##                                                     path:hsa00010 
##             "Glycolysis / Gluconeogenesis - Homo sapiens (human)" 
##                                                     path:hsa00020 
##                "Citrate cycle (TCA cycle) - Homo sapiens (human)" 
##                                                     path:hsa00030 
##                "Pentose phosphate pathway - Homo sapiens (human)" 
##                                                     path:hsa00040 
## "Pentose and glucuronate interconversions - Homo sapiens (human)" 
##                                                     path:hsa00051 
##          "Fructose and mannose metabolism - Homo sapiens (human)" 
##                                                     path:hsa00052 
##                     "Galactose metabolism - Homo sapiens (human)"
</code></pre></div></div>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Pull all genes for each pathway</span><span class="w">
</span><span class="n">pathway.codes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sub</span><span class="p">(</span><span class="s2">"path:"</span><span class="p">,</span><span class="w"> </span><span class="s2">""</span><span class="p">,</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">pathways.list</span><span class="p">))</span><span class="w">
</span><span class="n">genes.by.pathway</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">sapply</span><span class="p">(</span><span class="n">pathway.codes</span><span class="p">,</span><span class="w">
	</span><span class="k">function</span><span class="p">(</span><span class="n">pwid</span><span class="p">){</span><span class="w">
		</span><span class="n">pw</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">keggGet</span><span class="p">(</span><span class="n">pwid</span><span class="p">)</span><span class="w">
		</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">pw</span><span class="p">[[</span><span class="m">1</span><span class="p">]]</span><span class="o">$</span><span class="n">GENE</span><span class="p">))</span><span class="w"> </span><span class="nf">return</span><span class="p">(</span><span class="kc">NA</span><span class="p">)</span><span class="w">
		</span><span class="n">pw2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pw</span><span class="p">[[</span><span class="m">1</span><span class="p">]]</span><span class="o">$</span><span class="n">GENE</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="kc">TRUE</span><span class="p">,</span><span class="kc">FALSE</span><span class="p">)]</span><span class="w"> </span><span class="c1"># may need to modify this to c(FALSE, TRUE) for other organisms</span><span class="w">
		</span><span class="n">pw2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">unlist</span><span class="p">(</span><span class="n">lapply</span><span class="p">(</span><span class="n">strsplit</span><span class="p">(</span><span class="n">pw2</span><span class="p">,</span><span class="w"> </span><span class="n">split</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">";"</span><span class="p">,</span><span class="w"> </span><span class="n">fixed</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">),</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="n">x</span><span class="p">[</span><span class="m">1</span><span class="p">]))</span><span class="w">
		</span><span class="nf">return</span><span class="p">(</span><span class="n">pw2</span><span class="p">)</span><span class="w">
	</span><span class="p">}</span><span class="w">
</span><span class="p">)</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">genes.by.pathway</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## $hsa00010
##  [1] "10327"  "124"    "125"    "126"    "127"    "128"    "130"   
##  [8] "130589" "131"    "160287" "1737"   "1738"   "2023"   "2026"  
## [15] "2027"   "217"    "218"    "219"    "220"    "2203"   "221"   
## [22] "222"    "223"    "224"    "226"    "229"    "230"    "2538"  
## [29] "2597"   "26330"  "2645"   "2821"   "3098"   "3099"   "3101"  
## [36] "387712" "3939"   "3945"   "3948"   "441531" "501"    "5105"  
## [43] "5106"   "5160"   "5161"   "5162"   "5211"   "5213"   "5214"  
## [50] "5223"   "5224"   "5230"   "5232"   "5236"   "5313"   "5315"  
## [57] "55276"  "55902"  "57818"  "669"    "7167"   "80201"  "83440" 
## [64] "84532"  "8789"   "92483"  "92579"  "9562"  
## 
## $hsa00020
##  [1] "1431"  "1737"  "1738"  "1743"  "2271"  "3417"  "3418"  "3419" 
##  [9] "3420"  "3421"  "4190"  "4191"  "47"    "48"    "4967"  "50"   
## [17] "5091"  "5105"  "5106"  "5160"  "5161"  "5162"  "55753" "6389" 
## [25] "6390"  "6391"  "6392"  "8801"  "8802"  "8803" 
## 
## $hsa00030
##  [1] "132158" "2203"   "221823" "226"    "229"    "22934"  "230"   
##  [8] "2539"   "25796"  "2821"   "414328" "51071"  "5211"   "5213"  
## [15] "5214"   "5226"   "5236"   "55276"  "5631"   "5634"   "6120"  
## [22] "64080"  "6888"   "7086"   "729020" "8277"   "84076"  "8789"  
## [29] "9104"   "9563"  
## 
## $hsa00040
##  [1] "10327"  "10720"  "10941"  "231"    "27294"  "2990"   "51084" 
##  [8] "51181"  "54490"  "54575"  "54576"  "54577"  "54578"  "54579" 
## [15] "54600"  "54657"  "54658"  "54659"  "57016"  "574537" "6120"  
## [22] "6652"   "729020" "729920" "7358"   "7360"   "7363"   "7364"  
## [29] "7365"   "7366"   "7367"   "79799"  "9365"   "9942"  
## 
## $hsa00051
##  [1] "197258" "2203"   "226"    "229"    "230"    "231"    "26007" 
##  [8] "2762"   "29925"  "29926"  "3098"   "3099"   "3101"   "3795"  
## [15] "4351"   "5207"   "5208"   "5209"   "5210"   "5211"   "5213"  
## [22] "5214"   "5372"   "5373"   "55556"  "57016"  "57103"  "6652"  
## [29] "7167"   "7264"   "80201"  "8789"   "8790"  
## 
## $hsa00052
##  [1] "130589" "231"    "2538"   "2548"   "2582"   "2584"   "2592"  
##  [8] "2595"   "2645"   "2683"   "2717"   "2720"   "3098"   "3099"  
## [15] "3101"   "3906"   "3938"   "5211"   "5213"   "5214"   "5236"  
## [22] "55276"  "57016"  "57818"  "6476"   "7360"   "80201"  "8704"  
## [29] "8972"   "92579"  "93432"
</code></pre></div></div>

<p>Read in DE file to be used in enrichment testing:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">head</span><span class="p">(</span><span class="n">geneList</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##         2268        57188         3964         3744         6286 
## 6.496525e-09 1.440800e-08 2.552955e-08 4.106524e-08 4.463872e-08 
##         8659 
## 9.412370e-08
</code></pre></div></div>

<p><strong>2.</strong> Apply Wilcoxon rank-sum test to each pathway, testing if “in” p-values are smaller than “out” p-values:</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Wilcoxon test for each pathway</span><span class="w">
</span><span class="n">pVals.by.pathway</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">sapply</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">genes.by.pathway</span><span class="p">),</span><span class="w">
	</span><span class="k">function</span><span class="p">(</span><span class="n">pathway</span><span class="p">)</span><span class="w"> </span><span class="p">{</span><span class="w">
		</span><span class="n">pathway.genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">genes.by.pathway</span><span class="p">[[</span><span class="n">pathway</span><span class="p">]]</span><span class="w">
		</span><span class="n">list.genes.in.pathway</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">intersect</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">),</span><span class="w"> </span><span class="n">pathway.genes</span><span class="p">)</span><span class="w">
		</span><span class="n">list.genes.not.in.pathway</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">setdiff</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">geneList</span><span class="p">),</span><span class="w"> </span><span class="n">list.genes.in.pathway</span><span class="p">)</span><span class="w">
		</span><span class="n">scores.in.pathway</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">geneList</span><span class="p">[</span><span class="n">list.genes.in.pathway</span><span class="p">]</span><span class="w">
		</span><span class="n">scores.not.in.pathway</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">geneList</span><span class="p">[</span><span class="n">list.genes.not.in.pathway</span><span class="p">]</span><span class="w">
		</span><span class="k">if</span><span class="w"> </span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">scores.in.pathway</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">){</span><span class="w">
			</span><span class="n">p.value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">wilcox.test</span><span class="p">(</span><span class="n">scores.in.pathway</span><span class="p">,</span><span class="w"> </span><span class="n">scores.not.in.pathway</span><span class="p">,</span><span class="w"> </span><span class="n">alternative</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"less"</span><span class="p">)</span><span class="o">$</span><span class="n">p.value</span><span class="w">
		</span><span class="p">}</span><span class="w"> </span><span class="k">else</span><span class="p">{</span><span class="w">
			</span><span class="n">p.value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="kc">NA</span><span class="w">
		</span><span class="p">}</span><span class="w">
		</span><span class="nf">return</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">p.value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">p.value</span><span class="p">,</span><span class="w"> </span><span class="n">Annotated</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">length</span><span class="p">(</span><span class="n">list.genes.in.pathway</span><span class="p">)))</span><span class="w">
	</span><span class="p">}</span><span class="w">
</span><span class="p">))</span><span class="w">

</span><span class="c1"># Assemble output table</span><span class="w">
</span><span class="n">outdat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">pathway.code</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">pVals.by.pathway</span><span class="p">))</span><span class="w">
</span><span class="n">outdat</span><span class="o">$</span><span class="n">pathway.name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pathways.list</span><span class="p">[</span><span class="n">outdat</span><span class="o">$</span><span class="n">pathway.code</span><span class="p">]</span><span class="w">
</span><span class="n">outdat</span><span class="o">$</span><span class="n">p.value</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pVals.by.pathway</span><span class="p">[,</span><span class="s2">"p.value"</span><span class="p">]</span><span class="w">
</span><span class="n">outdat</span><span class="o">$</span><span class="n">Annotated</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pVals.by.pathway</span><span class="p">[,</span><span class="s2">"Annotated"</span><span class="p">]</span><span class="w">
</span><span class="n">outdat</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">outdat</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">outdat</span><span class="o">$</span><span class="n">p.value</span><span class="p">),]</span><span class="w">
</span><span class="n">head</span><span class="p">(</span><span class="n">outdat</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>##     pathway.code
## 179     hsa04612
## 329     hsa05332
## 188     hsa04650
## 328     hsa05330
## 242     hsa04940
## 327     hsa05323
##                                                         pathway.name
## 179       Antigen processing and presentation - Homo sapiens (human)
## 329                 Graft-versus-host disease - Homo sapiens (human)
## 188 Natural killer cell mediated cytotoxicity - Homo sapiens (human)
## 328                       Allograft rejection - Homo sapiens (human)
## 242                  Type I diabetes mellitus - Homo sapiens (human)
## 327                      Rheumatoid arthritis - Homo sapiens (human)
##          p.value Annotated
## 179 1.840993e-08        74
## 329 2.009811e-08        40
## 188 8.400982e-07       118
## 328 9.050700e-06        36
## 242 2.435849e-05        42
## 327 2.827471e-05        84
</code></pre></div></div>
<ul>
  <li>p.value: P-value for Wilcoxon rank-sum testing, testing that p-values from DE analysis for genes in the pathway are smaller than those not in the pathway</li>
  <li>Annotated: Number of genes in the pathway (regardless of DE p-value)</li>
</ul>

<p>The Wilcoxon rank-sum test is the nonparametric analogue of the two-sample t-test.  It compares the ranks of observations in two groups.  It is more powerful than the Kolmogorov-Smirnov test.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sessionInfo</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] Rgraphviz_2.28.0     org.Hs.eg.db_3.8.2   KEGGREST_1.24.0     
##  [4] topGO_2.36.0         SparseM_1.77         GO.db_3.8.2         
##  [7] AnnotationDbi_1.46.1 IRanges_2.18.1       S4Vectors_0.22.0    
## [10] Biobase_2.44.0       graph_1.62.0         BiocGenerics_0.30.0 
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.2         XVector_0.24.0     compiler_3.6.1    
##  [4] pillar_1.4.2       zlibbioc_1.30.0    tools_3.6.1       
##  [7] zeallot_0.1.0      digest_0.6.20      bit_1.1-14        
## [10] RSQLite_2.1.2      evaluate_0.14      memoise_1.1.0     
## [13] tibble_2.1.3       lattice_0.20-38    png_0.1-7         
## [16] pkgconfig_2.0.2    rlang_0.4.0        DBI_1.0.0         
## [19] curl_4.0           yaml_2.2.0         xfun_0.9          
## [22] httr_1.4.1         stringr_1.4.0      knitr_1.24        
## [25] Biostrings_2.52.0  vctrs_0.2.0        bit64_0.9-7       
## [28] R6_2.4.0           rmarkdown_1.15     blob_1.2.0        
## [31] magrittr_1.5       backports_1.1.4    htmltools_0.3.6   
## [34] matrixStats_0.54.0 stringi_1.4.3      crayon_1.3.4
</code></pre></div></div>


</div>



<script src="/assets/js/scale.fix.js"></script>



<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'true', 'auto');
    ga('send', 'pageview');
</script>


<script>


var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};
//var scrollpos=0;
//document.getElementById("mySidenav").onscroll = function() {scrollpos = document.getElementById("mySidenav").scrollTop;}

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
