
<link rel="stylesheet" href="/assets/highlight/styles/default.css">
<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;margin:5;background-color:#CFB53B; cursor:pointer;padding-right:50px">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="padding-left:50px;padding-top:10px;background-color:#CFB53B;">UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a>

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>

<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:160px">

<h1 id="closing-thoughts">Closing Thoughts</h1>

<p><strong>The forest through the trees.</strong></p>

<p><img src="base_figures/closing_figure1.png" alt="closing_figure1" width="800px" /></p>

<h2 id="be-consistent">Be Consistent</h2>

<ul>
  <li>Be Consistent</li>
  <li>Be Consistent</li>
  <li>Be Consistent</li>
  <li>Be Consistent</li>
  <li>Be Consistent</li>
</ul>

<h2 id="prerequisites-for-bioinformatics">Prerequisites for bioinformatics</h2>

<ul>
  <li>Access to a multi-core (24 cpu or greater), ‘high’ memory 64Gb or greater Linux server.</li>
  <li>Familiarity with the ’command line’ and at least one programming language.</li>
  <li>Basic knowledge of how to install software</li>
  <li>Basic knowledge of R (or equivalent) and statistical programming</li>
  <li>Basic knowledge of Statistics and model building</li>
</ul>

<h2 id="the-bottom-line">The bottom line</h2>

<ul>
  <li>Spend the time (and money) planning and producing good quality, accurate and sufficient data for your experiment.</li>
  <li>Get to know to your data, develop and test expectations</li>
  <li>Result, you’ll spend much less time (and less money) extracting biological significance and results during analysis.</li>
</ul>

<h2 id="workshop-week-2-reservation">Workshop Week 2 reservation</h2>

<p><strong>workshop       ACTIVE  2019-08-19T00:00:00  2019-08-31T00:00:00  12-00:00:00  rafter-[0,2,8,14,18]</strong></p>

<p>My recommendation is to follow all of the instructions again, from the beginning on your own and send emails to</p>

<p>bioinformatics.training@ucdavis.edu</p>

<p>And we will be responsive to answering questions.</p>

<h2 id="upcoming-workshops">Upcoming Workshops</h2>

<p>Now in our 12th year of training researchers, the UC Davis Bioinformatics Core invites you to attend one or more of our 2019 workshops. The following workshops are suitable for beginners, and are also of interest to those with some data analysis experience.</p>

<p>For more information on these and other events, and to register, go to https://registration.genomecenter.ucdavis.edu/</p>

<p><strong>September 3-6: Microbial Community Analysis @ UC Davis</strong></p>

<p>This workshop covers amplicon-based microbial analysis and methodology using high throughput sequencing technologies.  Topics will include:</p>

<ul>
  <li>Differences between community analysis with amplicons versus shotgun metagenomics and metatranscriptomics</li>
  <li>Experimental design</li>
  <li>Sequence data quality assurance and expectations</li>
  <li>Processing of amplicon reads (using the python application dbcAmplicons)</li>
  <li>Analysis of community structure (using R and vegan)</li>
  <li>Lecture on metagenome assembly and metatranscriptome assembly and annotation</li>
  <li>Post analysis and visualization (using R)</li>
</ul>

<p>Exercises will be performed with provided datasets, using command-line interaction and with RStudio.</p>

<p>For more information and to register, go to https://registration.genomecenter.ucdavis.edu/.</p>

<p><strong>December: Bioinformatics Prerequisites Workshop @ UC Davis</strong></p>

<p>This workshop will cover the basic prerequisites required before performing bioinformatics analysis, including basic command line and high performance computing concepts, how to install software, use help, run applications, and advanced command line (shell scripting, pipes, etc.). Also covered will be basic R programming, basic statistical concepts, and model building. There are no prerequisites for this workshop other than an interest in bioinformatics!</p>

<h2 id="contact-and-notifications">Contact and Notifications</h2>

<p>Questions about training?  Contact us at training.bioinformatics@ucdavis.edu</p>

<p>To subscribe / unsubscribe from the Bioinformatics Core email lists, please follow the instructions at the bottom of http://bioinformatics.ucdavis.edu/contact-us/.</p>


</div>



<script src="/assets/js/scale.fix.js"></script>



<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'true', 'auto');
    ga('send', 'pageview');
</script>


<script>


var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};
//var scrollpos=0;
//document.getElementById("mySidenav").onscroll = function() {scrollpos = document.getElementById("mySidenav").scrollTop;}

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
