
<link rel="stylesheet" href="/assets/highlight/styles/default.css">
<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;margin:5;background-color:#CFB53B; cursor:pointer;padding-right:50px">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="padding-left:50px;padding-top:10px;background-color:#CFB53B;">UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a>

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>

<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:160px">

<h3 id="the-menu-to-the-left-has-links-to-all-of-the-sections-and-talks-we-will-be-covering-this-week-you-can-use-the-menu-icon-in-the-upper-left-to-toggle-the-menu">The menu to the left has links to all of the sections and talks we will be covering this week. You can use the Menu icon in the upper left to toggle the menu.</h3>

<p>Bioinformatics: RNA-Seq Workshop @ UC Davis<br />
Aug. 19, 2019, 9 a.m. - Aug. 23, 2019, 4:30 p.m.<br />
Organizer - Bioinformatics Core<br />
Contact - UC Davis Bioinformatics Core, training.bioinformatics@ucdavis.edu</p>

<p>This workshop will include a rich collection of lectures and hands-on sessions, covering both theory and tools associated with command-line RNA-seq data analysis. Participants will explore experimental design, cost estimation, data generation, and analysis of RNA-Seq data generated on the Illumina sequencing platform. Participants will explore software and protocols, create and modify workflows, and diagnose/treat problematic data utilizing high performance computing services. Exercises will be performed with provided datasets, using command-line interaction.</p>

<p>A preliminary agenda for the week includes:</p>

<ul>
  <li>Monday	Introduction to the command line, cluster resources, and the R programming language (or advanced computing topics)</li>
  <li>Tuesday	Introduction to sequence data and preprocessing</li>
  <li>Wednesday        	Mapping data to a genome and transcript expression quantification</li>
  <li>Thursday	Differential expression analysis</li>
  <li>Friday	Post differential expression analysis, including gene set enrichment analysis, Gene Ontology/Pathway analysis, Generation of figures and tables</li>
</ul>

<p>Lectures throughout the week will include topics such as:</p>

<ul>
  <li>Sequencing technologies</li>
  <li>What is Bioinformatics?</li>
  <li>Hardware and software considerations</li>
  <li>Experimental design and cost estimation</li>
  <li>TAG-Seq vs. mRNA-Seq</li>
</ul>

<h3 id="industry-sponsors">Industry Sponsors</h3>

<table>
  <tbody>
    <tr>
      <td>Monday</td>
      <td><img src="base_figures/kapa.jpeg" alt="kapa" width="150px" /></td>
      <td> </td>
    </tr>
    <tr>
      <td>Monday Evening Social</td>
      <td><img src="base_figures/illumina.png" alt="illumina" width="150px" /></td>
      <td> </td>
    </tr>
    <tr>
      <td>Tuesday</td>
      <td><img src="base_figures/agilant.png" alt="agilant" width="150px" /></td>
      <td> </td>
    </tr>
    <tr>
      <td>Wednesday</td>
      <td><img src="base_figures/lexogen.jpeg" alt="lexogen" width="150px" /></td>
      <td> </td>
    </tr>
    <tr>
      <td>Thursday</td>
      <td><img src="base_figures/takara.png" alt="takara" width="150px" /></td>
      <td> </td>
    </tr>
    <tr>
      <td>Friday</td>
      <td><img src="base_figures/10x.png" alt="10x genomics" width="150px" /></td>
      <td> </td>
    </tr>
  </tbody>
</table>

<h3 id="faq">FAQ</h3>

<p>Who should attend? … Prior course participants have included faculty, post docs, grad students, staff, and industry researchers.</p>

<p>What are the prerequisites? … There are no prerequisites other than familiarity with general biological concepts and an enthusiasm for learning bioinformatics data analysis. Some familiarity with the command-line and R is desirable.  However, we will dedicate time on the first day to bringing everyone up-to-speed to be able to run the commands needed during this workshop.</p>

<p>What do I need to bring? … You will need to bring your own laptop to use, have a recent version of Java installed, and an application that will allow you to ssh into a server (e.g. Putty for windows, or iTerm for Apple) as well as the ability to connect to the internet via wifi. Computational exercises will be performed on a remote server. We can help with setup early on in the workshop. If your department does not allow you to install software on the laptop you’ll be bringing, please contact us ahead of time.</p>

<p>Can I bring my own data? … We will provide experimental datasets for use during the workshop, as this helps to keep the workshop moving. There will be time, however, to discuss your own datasets and how you might work with them outside of the workshop.</p>

<p>How much does it cost? … The registration fee is $1,500 for current students, postdocs, staff and faculty at UC campus’ or UC National Labs (LBNL, LANL, LLNL), $1,750 for other academic or non-profit participants and $2,500 for other participants. We accept credit cards, as well as UC Davis recharge accounts, for payment.  Registration fees include light breakfast, lunch, and snacks, but do not include dinner, lodging or parking fees.</p>

<p>Where is the workshop? … It will be held in Davis and will run from 9:00am to 4:30pm on the dates indicated. The specific location will be provided at a later date.</p>

<p>How do I apply? … All registration is “first-come, first-served”. There is no application process.  So, sign up as soon as possible to ensure your place in the workshop.</p>

<p>Where can I find more information, including your policies?  … Go to our website (bioinformatics.ucdavis.edu/training/) and check out our FAQ and Policies.</p>

<h3 id="questions">Questions</h3>

<p>If you have any questions, please don’t hesitate to contact us at training.bioinformatics@ucdavis.edu</p>

<h3 id="register">Register</h3>

<p>https://registration.genomecenter.ucdavis.edu/events/RNASeq-workshop_Aug_2019/</p>

<p><strong>Registration is closed for this event</strong></p>


</div>



<script src="/assets/js/scale.fix.js"></script>



<script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
    ga('create', 'true', 'auto');
    ga('send', 'pageview');
</script>


<script>


var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};
//var scrollpos=0;
//document.getElementById("mySidenav").onscroll = function() {scrollpos = document.getElementById("mySidenav").scrollTop;}

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
