
<head>
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-146433201-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-146433201-1');

</script>
</head>


<link rel="stylesheet" href="/assets/highlight/styles/default.css">

<script src="/assets/highlight/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>


<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/css/bootstrap.min.css">
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>

<title>UCD Bioinformatics Core RNA-Seq Workshop</title>

<meta name="viewport" content="width=device-width, initial-scale=1">

<link rel="shortcut icon" type="image/png" href="/assets/img/favicon.png">

<meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="chrome=1">
        <link rel="icon" type="image/png"  href="/assets/img/favicon.png">
        <title>UC Davis Bioinformatics Workshop Base Template by ucdavis-bioinformatics-training</title>

        <link rel="stylesheet" href="/assets/css/styles.css?v=22a1bf1e038521d63472a401b032dbdd184191bc">
    <meta name="viewport" content="width=device-width">

<body>
    <div style="background-color: #001A46" class="header sticky" id="myHeader" style="width:100%;">
      <a href="https://bioinformatics.ucdavis.edu/training/" class="logo">
          <img src="/assets/img/biocore_banner.png">
      </a>
        <div>
            <h2 style="float: left;padding-left:10px;padding-top:8px;background-color:#CFB53B; cursor:pointer;">
                <div onclick="doNav()">&#9776; Menu </div>
            </h2>
            <h2 style="border-radius: 0px;padding-top:10px;background-color:#CFB53B;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;UC Davis Bioinformatics Workshop Base Template</h2>
        </div>
    </div>
</body>


<div>
    <div id="mySidenav" class="sidenav">
      <!-- <a href="javascript:void(0)" class="closebtn" onclick="doNav()">&times;</a> -->

        <table style="padding-top: 50px">
            
                
                
                  <tr><th><a href="/">Home</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Introduction</th>
                  
                
                
                  <tr><th><a href="/welcome">Intro to the Workshop and Core</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/Genomics_a_perspective.pdf">What is Bioinformatics?</a></th></tr>
                
            
                
                
                  <tr><th><a href="/pdfs/ExperimentalDesign.pdf">Experimental Design and Cost Estimation</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Command-Line and the Cluster</th>
                  
                
                
                  <tr><th><a href="/cli/logging-in">Logging in and Transferring Files</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/command-line-intro">Intro to Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/advanced-command-line">Advanced Command-Line</a></th></tr>
                
            
                
                
                  <tr><th><a href="/cli/cluster">Running jobs on the Cluster and using modules</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Intro to R and Rstudio</th>
                  
                
                
                  <tr><th><a href="/intro2R/RStudio">Getting Started</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/Intro2R">Intro to R</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/data_in_R_prepare">Prepare Data in R (extra)</a></th></tr>
                
            
                
                
                  <tr><th><a href="/intro2R/orig_data_in_R">Data in R (extra)</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Data Reduction</th>
                  
                
                
                  <tr><th><a href="/data_reduction/filetypes">Files and Filetypes for RNA-Seq</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/project_setup">Project Setup</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/preproc_htstream">Preprocessing Data</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/alignment">Aligners</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/counts">Generating Summarized Counts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/data_reduction/salmon">Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Support</th>
                  
                
                
                  <tr><th><a href="/cheatSheetIndex">Cheat Sheets</a></th></tr>
                
            
                
                
                  <tr><th><a href="/software">Software and Links</a></th></tr>
                
            
                
                
                  <tr><th><a href="/script_links">Scripts</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">Differential Expression and Pathway Analysis</th>
                  
                
                
                  <tr><th><a href="/differential_expression/de_analysis_prepare">Prepare Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/annotation">Get Gene Annotation</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/DE_Analysis">Differential Expression Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/enrichment">Pathway Analysis</a></th></tr>
                
            
                
                
                  <tr><th><a href="/differential_expression/compare_star_salmon">Comparison between STAR and Salmon</a></th></tr>
                
            
                
                  <th style="font-size:22px; color:#CFB53B">ETC</th>
                  
                
                
                  <tr><th><a href="/closing">Closing thoughts</a></th></tr>
                
            
                
                
                  <tr><th><a href="/photos/photos">Workshop Photos</a></th></tr>
                
            
                
                
                  <tr id=Github ><th><a href="https://github.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop">Github</a></th></tr>
                
            
                
                
                  <tr id=Biocore website ><th><a href="http://bioinformatics.ucdavis.edu">Biocore website</a></th></tr>
                
            
        </table>
    </div>
</div>


<div id="main" style="padding-left: 50px; padding-right: 50px; margin-left: 300px; padding-top:170px">

<h1 id="running-jobs-on-the-cluster-and-using-modules">Running jobs on the cluster and using modules</h1>

<p><strong>1.</strong> In the UC Davis Bioinformatics Core we have a large computational cluster (named lssc0) that we use for our analyses. The job scheduling system we use on this cluster is called <a href="https://slurm.schedmd.com/">Slurm</a>. In this section, we will go through examples of the commands we will be using to interact with the cluster.</p>

<p>First, what is a cluster?</p>

<p><img src="figures/cluster_diagram.png" alt="figures/cluster_diagram" width="800px" /></p>

<p>The basic architecture of a compute cluster consists of a “head node”, which is the computer from which a user submits jobs to run, and “compute nodes”, which are a large number of computers on which the jobs can be run. It is also possible to log into a compute node and run jobs directly from there. <strong>In general you should never run a job directly on the head node!</strong></p>

<hr />
<p><strong>2.</strong> Now, let’s look at a few slurm commands.</p>

<p>The main commands we will be using are srun, sbatch, squeue, scancel, and sacct. First, log into the head node (tadpole.genomecenter.ucdavis.edu) and make a directory for yourself where you will be doing all your work.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mkdir /share/workshop/$USER
cd /share/workshop/$USER
</code></pre></div></div>

<p><strong>2a.</strong> <a href="https://slurm.schedmd.com/srun.html">‘srun’</a> is used to run a job interactively. We most often use it to start an interactive session on a compute node. Take a look at the options to srun:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>srun --help
</code></pre></div></div>

<p>Our cluster requires that you specify a time limit for your job. If your job exceeds these limits, then it will be killed. So try running the following to create an interactive session on a node:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>srun -t 00:30:00 -c 4 -n 1 --mem 2000 --partition production --account workshop --reservation workshop --pty /bin/bash
</code></pre></div></div>

<p>This command is requesting a compute node with a time limit of 30 minutes (-t), one processor (-c), a max memory of 2Gb [2000] (–mem), using a compute account and reservation for this workshop (an option you would not normally use), and then finally, specifying a shell to run in a terminal (“–pty” option). Run this command to get to a compute node when you want to run jobs on the command-line directly.</p>

<div class="output">srun: job 29390113 queued and waiting for resources
srun: job 29390113 has been allocated resources
groups: cannot find name for group ID 2020
bash: /home/msettles/.bashrc: Permission denied
msettles@drove-13:~$
</div>

<p>You safely ignore the working and error. Notice that we are no longer on tadpole, but are now on drove-13 in this example, one of our compute nodes.</p>

<p>use Exit on the command line to exit the session</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>exit
</code></pre></div></div>

<hr />
<p><strong>2b.</strong> <a href="https://slurm.schedmd.com/sbatch.html">‘sbatch’</a> is used to submit jobs to run on the cluster. Typically it is used to run many jobs via the scheduler non-interactively. Look at the options for sbatch:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sbatch --help
</code></pre></div></div>

<p>Generally, we do not use any options for sbatch … we typically give it a script (i.e. a text file with commands inside) to run. Let’s take a look at a template script <a href="/cli/templates/template.slurm">template.slurm</a>:</p>

<div class="script">#!/bin/bash

# options for sbatch
#SBATCH --job-name=name # Job name
#SBATCH --nodes=1 # should never be anything other than 1
#SBATCH --ntasks=1 # number of cpus to use
#SBATCH --time=60 # Acceptable time formats include "minutes", "minutes:seconds", "hours:minutes:seconds", "days-hours", "days-hours:minutes" and "days-hours:minutes:seconds".
#SBATCH --mem=2000 # Memory pool for all cores (see also --mem-per-cpu)
#SBATCH --partition=production # cluster partition
#SBATCH --reservation=workshop # cluster reservation
#SBATCH --account=workshop # cluster account to use for the job
##SBATCH --array=1-16 # Task array indexing, see https://slurm.schedmd.com/job_array.html, the double # means this line is commented out
#SBATCH --output=stdout.out # File to which STDOUT will be written
#SBATCH --error=stderr.err # File to which STDERR will be written
#SBATCH --mail-type=ALL
#SBATCH --mail-user=myemail@email.com

# for calculating the amount of time the job takes and echo the hostname
begin=`date +%s`
echo $HOSTNAME

# Sleep for 60 seconds
sleep 300

# getting end time to calculate time elapsed
end=`date +%s`
elapsed=`expr $end - $begin`
echo Time taken: $elapsed
</div>

<p>The first line tells sbatch what scripting language (bash here) the rest of the file is in. Any line that begins with a “#” symbol is ignored by the bash interpreter, those lines that begin with “#SBATCH” are used by the slurm controller. Those lines are for specifying sbatch options without having to type them on the command-line every time. In this script, on the next set of lines, we’ve put some code for calculating the time elapsed for the job and then we simply wait for 5 minutes (300 seconds) and exit. Lets try running it</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>cd /share/workshop/$USER
wget https://raw.githubusercontent.com/ucdavis-bioinformatics-training/2019_August_UCD_mRNAseq_Workshop/master/templates/template.slurm
cat template.slurm
sbatch template.slurm
</code></pre></div></div>

<p>The non slurm version is the <a href="/cli/templates/template.sh">template.sh</a> script. You’ll notice it looks the same only missing the #SBATCH commands.</p>

<p>After finishing you will see two new files in the directory stdout.out and stderr.err where stdout and stderr (respectively) were redirected to.</p>

<hr />
<p><strong>2c.</strong> <a href="https://slurm.schedmd.com/squeue.html">‘squeue’</a> is to list your currently queued/running jobs. T</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>squeue --help
</code></pre></div></div>

<p>Looking at the help documentation, we see that we can filter the results based on a number of criteria. The most useful option is “-u”, which you can use to see just the jobs for a particular user ID. The first column gives you the job ID of the job, the second is the partition (different queues for different types of machines), the name of the job, the user who ran the job, the state of the job (R is for running), the length of time the job has been running, the number of nodes the job is using, and finally, the node name where the job is running or a reason why the job is waiting.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>squeue -u username
</code></pre></div></div>

<div class="output">msettles@tadpole:/share/workshop/msettles$ squeue -u msettles
             JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
          29390121 productio     name msettles  R       0:06      1 drove-13
</div>

<p>You can see the job has been running (ST=R) for 6 seconds (TIME=0:06) on node drove-12. The jobid (here 29390121) can be used to cancel the job later, or get additional job information.</p>

<p><strong>2d.</strong> ‘scancel’ command is used to cancel jobs (either running or in queue).</p>

<p>You can give it a job ID, or if you use the “-u” option with your username, you can cancel all of your jobs at once.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>scancel -j 29390121
</code></pre></div></div>

<p>will cancel the above job if its still running.</p>

<p><strong>2e.</strong> ‘sacct’ command is used to get accounting data for any job that has ever run, using the job ID.</p>

<p>To get statistics on completed jobs by jobID (replace jobid with your jobid):</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sacct -j jobid --format=JobID,JobName,MaxRSS,Elapsed
</code></pre></div></div>

<p>To view the same information for all jobs of a user (replace username with your username):</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sacct -u username --format=JobID,JobName,MaxRSS,Elapsed
</code></pre></div></div>

<p><strong>You can get more information about each command by typing “<command /> –help” or by looking at <a href="https://slurm.schedmd.com/pdfs/summary.pdf">this summary page</a>.</strong></p>

<hr />
<h2 id="environment-modules">Environment modules</h2>

<p><strong>1.</strong> The ‘module’ command and its sub-commands. You will NOT find the ‘module’ command on all linux computers. Using modules is generally something that is used on a shared system (like clusters) and generally installed by the system’s administrator. The module system allows to only ‘load’ software when needed as well as to make multiple versions of software available at the same time. It basically changes your PATH variable (and possibly other environment variables) so that the shell searches the correct directories for the software you want to use. First, take a look at all the software available on our system:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module avail
</code></pre></div></div>

<p><img src="figures/modules_figure1.png" alt="modules_figure1" width="800px" /></p>

<p>This is a list of all the software (with different versions) that you can access. The format of ‘modules’ is software/version here you can see htstream has 3 different versions installed with the latest being 1.0.0. When you load a module without specifying a version, it will load the default (generally the latest) version. If you need an older version, you need to use the cooresponding version number:</p>

<p>Now try running the ‘hts_Stats’ app from htstream:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>hts_Stats
</code></pre></div></div>

<div class="output">msettles@tadpole:/share/workshop/msettles$hts_Stats
-bash: hts_Stats: command not found
</div>

<p>You should get an error saying that the command was not found. Take a look at your PATH variable.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>echo $PATH
</code></pre></div></div>

<div class="output">msettles@tadpole:/share/workshop/msettles$ echo $PATH
/software/slurm/17.11.2/lssc0-linux/sbin:/software/slurm/17.11.2/lssc0-linux/bin:/software/modules/1.923/lssc0-linux/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/puppetlabs/bin
</div>

<p>These are the directories (colon separated) that are searched for executable applications to run on the command-line. In order to access a piece of software that is not in one of these default directories, we need to use the ‘module load’ command, or set the PATH to locate it:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load htstream/1.0.0
hts_Stats
</code></pre></div></div>

<p>Use the ‘which’ command to find out where the ‘hts_Stats’ command is actually located:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>which hts_Stats
</code></pre></div></div>

<p>You’ll see that hts_Stats is located in a completely different place and yet you are able to access it. This is because the module command changes your PATH variable so that it has the correct directory. Take a look at your PATH again:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>echo $PATH
</code></pre></div></div>

<p>You’ll see that the directory for scythe has been added to PATH.</p>

<hr />
<p><strong>2.</strong> A few more module sub-commands that are useful:</p>

<p>‘module list’ will list all of your currently loaded modules in this terminal/session.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module list
</code></pre></div></div>

<div class="output">msettles@tadpole:/share/workshop/msettles$ module list
Currently Loaded Modulefiles:
 1) slurm/latest   2) htstream/1.0.0
</div>

<p><em>‘module unload’</em> will unload the module(s) you specify.</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>module load star
module load samtools
module list
module unload star
module list
</code></pre></div></div>

<div class="output">msettles@tadpole:/share/workshop/msettles$ module load star
Module star-2.7.0e-lssc0-linux loaded. STAR (Spliced Transcripts Alignment to a Reference) is an RNA-seq data aligner. NOTE: Indices must be indexed using this version or newer, they cannot be from a previous version.
msettles@tadpole:/share/workshop/msettles$ module load samtools
Module samtools-1.9-lssc0-linux loaded. Samtools is a suite of programs for interacting with high-throughput sequencing data.
msettles@tadpole:/share/workshop/msettles$ module list
Currently Loaded Modulefiles:
 1) slurm/latest   2) htstream/1.0.0   3) star/2.7.0e   4) samtools/1.9
msettles@tadpole:/share/workshop/msettles$ module unload star
msettles@tadpole:/share/workshop/msettles$ module list
Currently Loaded Modulefiles:
 1) slurm/latest   2) htstream/1.0.0   3) samtools/1.9
</div>

<p>‘module purge’ will unload all of your modules. Which simply means that it will take out the directories for all of the modules from your PATH variable. Take a look at $PATH now:</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>echo $PATH
module purge
module list
echo $PATH
</code></pre></div></div>

<div class="output">msettles@tadpole:/share/workshop/msettles$     echo $PATH
/software/samtools/1.9/lssc0-linux/bin:/software/htstream/1.0.0/lssc0-linux/bin:/share/biocore/software/bin:/software/modules/1.923/lssc0-linux/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/puppetlabs/bin
msettles@tadpole:/share/workshop/msettles$     module purge
msettles@tadpole:/share/workshop/msettles$     module list
No Modulefiles Currently Loaded.
msettles@tadpole:/share/workshop/msettles$     echo $PATH
/share/biocore/software/bin:/software/modules/1.923/lssc0-linux/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/opt/puppetlabs/bin
</div>


</div>



<script src="/assets/js/scale.fix.js"></script>

<script>
var navopen=1;

function doNav() {
  if (navopen == 1) {
    document.getElementById("mySidenav").style.width = "0px";
    document.getElementById("main").style.marginLeft = "0px";
    document.getElementById("myHeader").style.marginLeft = "0px";
    document.getElementById("mySidenav").style.overflowY = "hidden";
    navopen = 0;
  } else {
    document.getElementById("mySidenav").style.width = "300px";
    document.getElementById("main").style.marginLeft = "300px";
    document.getElementById("myHeader").style.marginLeft = "300px";
    document.getElementById("mySidenav").style.overflowY = "visible";
    navopen = 1;
  }
}

// When the user scrolls the page, execute myFunction
window.onscroll = function() {myFunction()};

// store the sidenav scroll pos for loading or set the pos
sn = document.getElementById("mySidenav");
sn.onscroll = function() {sessionStorage.setItem("scrollpos",sn.scrollTop);};
window.onload = function() {sn.scrollTop = sessionStorage.getItem("scrollpos");};

// Get the header
var header = document.getElementById("myHeader");

// Get the offset position of the navbar
var sticky = header.offsetTop;


function myFunction() {
  var x = document.getElementById("myDIV");
  if (x.style.display === "none") {
    x.style.display = "block";
  } else {
    x.style.display = "none";
  }
}

</script>
